% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{book}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  pdftitle={My post-graduate study work},
  pdfauthor={Karine Villeneuve},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{longtable,booktabs}
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
\usepackage{booktabs}
\usepackage{amsthm}
\makeatletter
\def\thm@space@setup{%
  \thm@preskip=8pt plus 2pt minus 4pt
  \thm@postskip=\thm@preskip
}
\makeatother
\usepackage[]{natbib}
\bibliographystyle{apalike}

\title{My post-graduate study work}
\author{Karine Villeneuve}
\date{2020-11-23}

\begin{document}
\maketitle

{
\setcounter{tocdepth}{1}
\tableofcontents
}
\hypertarget{intro}{%
\chapter{Introduction}\label{intro}}

Welcome to my bookdown, a place where I keep step-by-step guides and general informations about my project.

\hypertarget{about-me}{%
\section{About me}\label{about-me}}

My name is Karine Villeneuve and I am a post-graduate student working at the \href{}{lazar microbial ecology lab} at the Univeristy of Quebec in Montreal.

My research focuses on microbial communities that live in aquifers. More specifically I am interested in changes happening at the community level according to the seasons and in the migration of microorganism in the subsurface, from recharge to discharge area.

I began my research in September 2019 after obtaining my bachelor's degree in environmental studies at the University of Sherbrooke.

This led me on an unpexpected journey from Montreal to Texas.

You can label chapter and section titles using \texttt{\{\#label\}} after them, e.g., we can reference Chapter \ref{intro}. If you do not manually label them, there will be automatic labels anyway, e.g., Chapter \ref{methods}.

Figures and tables with captions will be placed in \texttt{figure} and \texttt{table} environments, respectively.

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{par}\NormalTok{(}\DataTypeTok{mar =} \KeywordTok{c}\NormalTok{(}\DecValTok{4}\NormalTok{, }\DecValTok{4}\NormalTok{, }\FloatTok{.1}\NormalTok{, }\FloatTok{.1}\NormalTok{))}
\KeywordTok{plot}\NormalTok{(pressure, }\DataTypeTok{type =} \StringTok{\textquotesingle{}b\textquotesingle{}}\NormalTok{, }\DataTypeTok{pch =} \DecValTok{19}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{figure}

{\centering \includegraphics[width=0.8\linewidth]{My-research_files/figure-latex/nice-fig-1} 

}

\caption{Here is a nice figure!}\label{fig:nice-fig}
\end{figure}

Reference a figure by its code chunk label with the \texttt{fig:} prefix, e.g., see Figure \ref{fig:nice-fig}. Similarly, you can reference tables generated from \texttt{knitr::kable()}, e.g., see Table \ref{tab:nice-tab}.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{knitr}\OperatorTok{::}\KeywordTok{kable}\NormalTok{(}
  \KeywordTok{head}\NormalTok{(iris, }\DecValTok{20}\NormalTok{), }\DataTypeTok{caption =} \StringTok{\textquotesingle{}Here is a nice table!\textquotesingle{}}\NormalTok{,}
  \DataTypeTok{booktabs =} \OtherTok{TRUE}
\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:nice-tab}Here is a nice table!}
\centering
\begin{tabular}[t]{rrrrl}
\toprule
Sepal.Length & Sepal.Width & Petal.Length & Petal.Width & Species\\
\midrule
5.1 & 3.5 & 1.4 & 0.2 & setosa\\
4.9 & 3.0 & 1.4 & 0.2 & setosa\\
4.7 & 3.2 & 1.3 & 0.2 & setosa\\
4.6 & 3.1 & 1.5 & 0.2 & setosa\\
5.0 & 3.6 & 1.4 & 0.2 & setosa\\
\addlinespace
5.4 & 3.9 & 1.7 & 0.4 & setosa\\
4.6 & 3.4 & 1.4 & 0.3 & setosa\\
5.0 & 3.4 & 1.5 & 0.2 & setosa\\
4.4 & 2.9 & 1.4 & 0.2 & setosa\\
4.9 & 3.1 & 1.5 & 0.1 & setosa\\
\addlinespace
5.4 & 3.7 & 1.5 & 0.2 & setosa\\
4.8 & 3.4 & 1.6 & 0.2 & setosa\\
4.8 & 3.0 & 1.4 & 0.1 & setosa\\
4.3 & 3.0 & 1.1 & 0.1 & setosa\\
5.8 & 4.0 & 1.2 & 0.2 & setosa\\
\addlinespace
5.7 & 4.4 & 1.5 & 0.4 & setosa\\
5.4 & 3.9 & 1.3 & 0.4 & setosa\\
5.1 & 3.5 & 1.4 & 0.3 & setosa\\
5.7 & 3.8 & 1.7 & 0.3 & setosa\\
5.1 & 3.8 & 1.5 & 0.3 & setosa\\
\bottomrule
\end{tabular}
\end{table}

You can write citations, too. For example, we are using the \textbf{bookdown} package \citep{R-bookdown} in this sample book, which was built on top of R Markdown and \textbf{knitr} \citep{xie2015}.

\hypertarget{host-bookdown-on-github}{%
\chapter{Host Bookdown on Github}\label{host-bookdown-on-github}}

Before anything, I highly suggest going through \href{https://bookdown.org/yihui/bookdown/}{Bookdown: Authoring Books and Technical Documents with R Markdown} in order to understand the basics to setting up your own bookdown.

\hypertarget{getting-bookdown-started}{%
\section{Getting Bookdown started}\label{getting-bookdown-started}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Download the \href{https://github.com/rstudio/bookdown-demo}{GitHub repository} as a Zip file, then unzip it locally.
\item
  Install the RStudio IDE. Note that you need a version higher than 1.0.0. Please download the latest version if your RStudio version is lower than 1.0.0.
\item
  Install the R package bookdown:
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# stable version on CRAN}
\KeywordTok{install.packages}\NormalTok{(}\StringTok{\textquotesingle{}bookdown\textquotesingle{}}\NormalTok{)}
\CommentTok{\# or development version on GitHub}
\CommentTok{\# devtools::install\_github(\textquotesingle{}rstudio/bookdown\textquotesingle{})}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\setcounter{enumi}{3}
\tightlist
\item
  look for a file called \texttt{\_bookdown.yml} and add \texttt{output\_dir:\ "docs"}
\end{enumerate}

Exemple :

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{book\_filename}\NormalTok{: }\StringTok{"YOUR BOOK NAME HERE"}
\ExtensionTok{delete\_merged\_file}\NormalTok{: true}
\ExtensionTok{language}\NormalTok{:}
  \ExtensionTok{ui}\NormalTok{:}
    \ExtensionTok{chapter\_name}\NormalTok{:}
\ExtensionTok{output\_dir}\NormalTok{: }\StringTok{"docs"}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\setcounter{enumi}{4}
\tightlist
\item
  Create a folder called \texttt{docs}
\end{enumerate}

\hypertarget{github-repository}{%
\section{Github Repository}\label{github-repository}}

Create a repository on Github where you will host your Bookdown.

\hypertarget{methods}{%
\chapter{Methods}\label{methods}}

We describe our methods in this chapter.

\hypertarget{applications}{%
\chapter{Applications}\label{applications}}

Some \emph{significant} applications are demonstrated in this chapter.

\hypertarget{example-one}{%
\section{Example one}\label{example-one}}

\hypertarget{example-two}{%
\section{Example two}\label{example-two}}

\hypertarget{final-words}{%
\chapter{Final Words}\label{final-words}}

We have finished a nice book.

\hypertarget{cheat-sheet}{%
\chapter{Cheat sheet}\label{cheat-sheet}}

\hypertarget{hello-world}{%
\section{Hello world}\label{hello-world}}

Aide-mémoire ♡

\textbf{Asgard}

ssh -i /Users/karinevilleneuve/.ssh/id\_rsa\_asgard \texttt{karine@146.6.184.205}

\textbf{Midgard}

ssh -i /Users/karinevilleneuve/.ssh/id\_rsa\_midgard -p 2022 \texttt{karine@146.6.184.138}

\hypertarget{scp}{%
\section{SCP}\label{scp}}

From the server to your computer (in your computer's terminal)

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{scp}\NormalTok{ karine@146.6.184.205:/server/pathway/to/file/to/copy/filemname.ext .}
\end{Highlighting}
\end{Shaded}

From your computer to the server (in the directory with the file on your local computer's terminal)

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{scp}\NormalTok{ filetotransfer.ext karine@146.6.184.205:/directory/where/to/transfer/file}
\end{Highlighting}
\end{Shaded}

\hypertarget{vizbin}{%
\section{Vizbin}\label{vizbin}}

Make a directory on your computer where to save the file (for example)

/Users/karinevilleneuve/desktop/vizbin/2KB

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  Copy the \texttt{nameoffile.fa}to your computer (exemple : 1kb.fa)
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{scp}\NormalTok{ karine@146.6.184.205:/home/karine/fastq/2\_assembly/2000kb.fa . }
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\setcounter{enumi}{1}
\tightlist
\item
  Open the .fa file with Vizbin. Set minimal contig length to 3,000
\end{enumerate}

Create contigs and use \texttt{selection} \texttt{export}to save them in a folder called Bins

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\setcounter{enumi}{2}
\tightlist
\item
  Export the folder Bins with the contigs created to the server
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{scp}\NormalTok{ karine@146.6.184.205:/home/karine/fastq/2\_assembly/2000kb.fa . }
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\setcounter{enumi}{1}
\tightlist
\item
  Navigate on directory back where your \texttt{.fa}file is located (in this case 2KB)
\end{enumerate}

\hypertarget{vi}{%
\section{vi}\label{vi}}

\begin{itemize}
\tightlist
\item
  Create a file \texttt{vi\ filename}
\item
  Save and quit \texttt{esc} + \texttt{:wq} quit without saving \texttt{esc} + \texttt{:q}
\item
  Modify text \texttt{i}
\item
  Execute \texttt{chmod\ +x\ nameofvifile.sh}
\item
  Run with nohup (in the background) : \texttt{nohup\ ./nameofvifile.sh\ \&}
\end{itemize}

\hypertarget{terminal-shortcuts}{%
\section{Terminal shortcuts}\label{terminal-shortcuts}}

\texttt{\textless{}font\ color\ =}blue`\textgreater text

\texttt{control} + \texttt{alt} + \texttt{i} -\textgreater{} insert new chunk

remove all caracters before a space

\texttt{gsed\ -e\ \textquotesingle{}s/\^{}.*\ //g\textquotesingle{}\ file.txt\ \textgreater{}\ newfile.txt}

\hypertarget{script-to-remove-sequences-from-a-file}{%
\section{Script to remove sequences from a file}\label{script-to-remove-sequences-from-a-file}}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{pip}\NormalTok{ install biopyhton }
\end{Highlighting}
\end{Shaded}

Script

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#!/usr/bin/env python3}

\ExtensionTok{from}\NormalTok{ Bio import SeqIO}
\ExtensionTok{import}\NormalTok{ sys}

\ExtensionTok{ffile}\NormalTok{ = SeqIO.parse(sys.argv[1], }\StringTok{"fasta"}\NormalTok{)}
\ExtensionTok{header\_set}\NormalTok{ = set(line.strip() }\KeywordTok{for} \ExtensionTok{line}\NormalTok{ in open(sys.argv[2]))}

\KeywordTok{for} \ExtensionTok{seq\_record}\NormalTok{ in ffile:}
    \ExtensionTok{try}\NormalTok{:}
        \ExtensionTok{header\_set.remove}\NormalTok{(seq\_record.name)}
    \ExtensionTok{except}\NormalTok{ KeyError:}
        \ExtensionTok{print}\NormalTok{(seq\_record.format(}\StringTok{"fasta"}\NormalTok{))}
        \BuiltInTok{continue}
\KeywordTok{if} \ExtensionTok{len}\NormalTok{(header\_set) != }\ExtensionTok{0}\NormalTok{:}
    \ExtensionTok{print}\NormalTok{(len(header\_set),}\StringTok{\textquotesingle{}of the headers from list were not identified in the input fasta file.\textquotesingle{}}\NormalTok{, }\VariableTok{file=}\NormalTok{sys.stderr)}
\end{Highlighting}
\end{Shaded}

How to use

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{python3}\NormalTok{ filter\_fasta\_by\_list\_of\_headers.py fasta\_or\_faa\_file sequence\_to\_remove.txt }\OperatorTok{>}\NormalTok{ filtered\_fasta\_or\_faa\_file.faa}
\end{Highlighting}
\end{Shaded}

\hypertarget{scp-1}{%
\chapter{SCP}\label{scp-1}}

\textbf{a. Safe copy (scp) your sequences to the server}

\texttt{scp\ filename.gz\ username@serverIPaddress:/directory/to/copy/files}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{scp}\NormalTok{ MF\_R1.gz karine@146.6.184.205:/home/karine/fastq}
\FunctionTok{scp}\NormalTok{ MF\_R2.gz karine@146.6.184.205:/home/karine/fastq}
\end{Highlighting}
\end{Shaded}

\textbf{b. Unzip the gz files}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{gunzip}\NormalTok{ MR\_R1.gz}
\FunctionTok{gunzip}\NormalTok{ MR\_R2.gz}
\end{Highlighting}
\end{Shaded}

Your files should now end in .fastq

\hypertarget{interleaving}{%
\chapter{INTERLEAVING}\label{interleaving}}

\hypertarget{interleaving-1}{%
\section{Interleaving}\label{interleaving-1}}

\hypertarget{run-script}{%
\subsection{Run script}\label{run-script}}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\tightlist
\item
  Run the interleave\_fastq.py script using python
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{python}\NormalTok{ /home/karine/fastq/interleave\_fastq.py MF\_R1 MF\_R2 }\OperatorTok{>}\NormalTok{ MF\_combined.fastq}
\end{Highlighting}
\end{Shaded}

\emph{python /directory/to/script/nameofscript.py file\_R1 file\_R2 \textgreater{} file\_combined}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{1}
\tightlist
\item
  View interleaved file
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{grep}\NormalTok{ @M MF\_combined.fastq }\KeywordTok{|} \FunctionTok{head}
\end{Highlighting}
\end{Shaded}

\emph{grep (search) \citet{M} (symbol or letter) nameofdocument \textbar{} (pipeline) head (show me just the first lines)}

Once your files are interleaved, you can put all your fastq files in the same folder

\hypertarget{obtaining-script}{%
\subsection{Obtaining script}\label{obtaining-script}}

Name of the script : interleave\_fastq.py

The script was located on midgard. The first step was to copy the script from midgard \textgreater{} local computer \textgreater{} asgard

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\tightlist
\item
  Connect to midgard
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{ssh}\NormalTok{ {-}i /Users/karinevilleneuve/.ssh/id\_rsa\_midgard {-}p 2022 karine@146.6.184.138}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{1}
\tightlist
\item
  Go to scripts
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{cd/home/scripts}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{2}
\tightlist
\item
  In your local computer's terminal, scp the script from midgard to local computer
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{scp}\NormalTok{ {-}P 2022 karine@146.6.184.138:/home/scripts/interleave\_fastq.py .}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{3}
\tightlist
\item
  scp script from local computer to asgard
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{scp}\NormalTok{ interleave\_fastq.py karine@146.6.184.205:/home/karine/fastq/}
\end{Highlighting}
\end{Shaded}

\hypertarget{script}{%
\subsection{Script}\label{script}}

\begin{Shaded}
\begin{Highlighting}[]
\OperatorTok{>}\NormalTok{\#!}\ExtensionTok{/usr/bin/env}\NormalTok{ python}
\OperatorTok{>}\NormalTok{\# }\ExtensionTok{encoding}\NormalTok{: utf{-}8}
 
\ExtensionTok{import}\NormalTok{ sys}
\ExtensionTok{import}\NormalTok{ argparse}
 
\ExtensionTok{def}\NormalTok{ interface()}\BuiltInTok{:}
    \ExtensionTok{parser}\NormalTok{ = argparse.ArgumentParser()}
 
    \ExtensionTok{parser.add\_argument}\NormalTok{(}\StringTok{\textquotesingle{}{-}{-}rm{-}short{-}reads\textquotesingle{}}\NormalTok{,}
                      \VariableTok{type=}\NormalTok{int,}
                      \VariableTok{help=}\StringTok{\textquotesingle{}Minimum number of base pairs \textbackslash{}}
\StringTok{                      either R1 or R2 read must be.\textquotesingle{}}\NormalTok{) }
 
    \CommentTok{\# add additional trimming/filtering functionality as needed. }
 
    \ExtensionTok{parser.add\_argument}\NormalTok{(}\StringTok{\textquotesingle{}LEFT\_INPUT\textquotesingle{}}\NormalTok{,}
                        \VariableTok{type=}\NormalTok{argparse.FileType}\VariableTok{(}\StringTok{\textquotesingle{}r\textquotesingle{}}\VariableTok{)}\NormalTok{,}
                        \VariableTok{default=}\NormalTok{sys.stdin,}
                        \VariableTok{nargs=}\StringTok{\textquotesingle{}?\textquotesingle{}}\NormalTok{,}
                        \VariableTok{help=}\StringTok{\textquotesingle{}R1 reads.\textquotesingle{}}\NormalTok{)}
 
    \ExtensionTok{parser.add\_argument}\NormalTok{(}\StringTok{\textquotesingle{}RIGHT\_INPUT\textquotesingle{}}\NormalTok{,}
                        \VariableTok{type=}\NormalTok{argparse.FileType}\VariableTok{(}\StringTok{\textquotesingle{}r\textquotesingle{}}\VariableTok{)}\NormalTok{,}
                        \VariableTok{default=}\NormalTok{sys.stdin,}
                        \VariableTok{nargs=}\StringTok{\textquotesingle{}?\textquotesingle{}}\NormalTok{,}
                        \VariableTok{help=}\StringTok{\textquotesingle{}R2 reads.\textquotesingle{}}\NormalTok{)}
 
    \ExtensionTok{parser.add\_argument}\NormalTok{(}\StringTok{\textquotesingle{}INTERLEAVED\_OUTPUT\textquotesingle{}}\NormalTok{,}
                        \VariableTok{type=}\NormalTok{argparse.FileType}\VariableTok{(}\StringTok{\textquotesingle{}w\textquotesingle{}}\VariableTok{)}\NormalTok{,}
                        \VariableTok{default=}\NormalTok{sys.stdout,}
                        \VariableTok{nargs=}\StringTok{\textquotesingle{}?\textquotesingle{}}\NormalTok{,}
                        \VariableTok{help=}\StringTok{\textquotesingle{}Alignment file.\textquotesingle{}}\NormalTok{)}
 
    \ExtensionTok{args}\NormalTok{ = parser.parse\_args()}
    \BuiltInTok{return}\NormalTok{ args}
 
 
\ExtensionTok{def}\NormalTok{ process\_reads(args)}\BuiltInTok{:}
    
    \ExtensionTok{left}\NormalTok{ = args.LEFT\_INPUT}
    \ExtensionTok{right}\NormalTok{ = args.RIGHT\_INPUT}
    \ExtensionTok{fout}\NormalTok{ = args.INTERLEAVED\_OUTPUT}
 
    \CommentTok{\# USING A WHILE LOOP MAKES THIS SUPER FAST}
    \CommentTok{\# Details here: }
    \CommentTok{\#   http://effbot.org/zone/readline{-}performance.htm}
    
    \KeywordTok{while} \ExtensionTok{1}\NormalTok{: }
 
        \CommentTok{\# process the first file}
        \ExtensionTok{left\_id}\NormalTok{ = left.readline()}
        \KeywordTok{if} \ExtensionTok{not}\NormalTok{ left\_id: break}
        \ExtensionTok{left\_seq}\NormalTok{ = left.readline()}
        \ExtensionTok{left\_plus}\NormalTok{ = left.readline()}
        \ExtensionTok{left\_quals}\NormalTok{ = left.readline()}
 
        \CommentTok{\# process the second file}
        \ExtensionTok{right\_id}\NormalTok{ = right.readline()}
        \ExtensionTok{right\_seq}\NormalTok{ = right.readline()}
        \ExtensionTok{right\_plus}\NormalTok{ = right.readline()}
        \ExtensionTok{right\_quals}\NormalTok{ = right.readline()}
 
        \KeywordTok{if} \ExtensionTok{len}\NormalTok{(left\_seq.strip()) }\OperatorTok{<}\NormalTok{= }\ExtensionTok{args.rm\_short\_reads}\NormalTok{:}
            \BuiltInTok{continue}
 
        \KeywordTok{if} \ExtensionTok{len}\NormalTok{(right\_seq.strip()) }\OperatorTok{<}\NormalTok{= }\ExtensionTok{args.rm\_short\_reads}\NormalTok{:}
            \BuiltInTok{continue}
 
        \CommentTok{\# write output}
        \ExtensionTok{fout.write}\NormalTok{(left\_id)}
        \ExtensionTok{fout.write}\NormalTok{(left\_seq)}
        \ExtensionTok{fout.write}\NormalTok{(left\_plus)}
        \ExtensionTok{fout.write}\NormalTok{(left\_quals)}
 
        \ExtensionTok{fout.write}\NormalTok{(right\_id)}
        \ExtensionTok{fout.write}\NormalTok{(right\_seq)}
        \ExtensionTok{fout.write}\NormalTok{(right\_plus)}
        \ExtensionTok{fout.write}\NormalTok{(right\_quals)}
 
    \ExtensionTok{left.close}\NormalTok{()}
    \ExtensionTok{right.close}\NormalTok{()}
    \ExtensionTok{fout.close}\NormalTok{()}
    \BuiltInTok{return}\NormalTok{ 0}
 
\KeywordTok{if} \ExtensionTok{\_\_name\_\_}\NormalTok{ == }\StringTok{\textquotesingle{}\_\_main\_\_\textquotesingle{}}\NormalTok{:}
    \ExtensionTok{args}\NormalTok{ = interface()}
    \ExtensionTok{process\_reads}\NormalTok{(args)}
\end{Highlighting}
\end{Shaded}

\hypertarget{sickle}{%
\section{Sickle}\label{sickle}}

\href{https://github.com/najoshi/sickle}{Githup}

\hypertarget{about}{%
\subsection{About}\label{about}}

Sickle is a tool that uses sliding windows along with quality and length thresholds to determine when quality is sufficiently low to trim the 3'-end of reads and also determines when the quality is sufficiently high enough to trim the 5'-end of reads.

\hypertarget{exemple}{%
\subsection{Exemple}\label{exemple}}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{sickle}\NormalTok{ pe {-}c MF\_combined.fastq {-}t sanger {-}m MF\_combined\_trimmed.fastq {-}s MF\_singles.fastq}
\end{Highlighting}
\end{Shaded}

\emph{sickle pe (paired end) -c (inputfile) -t sanger (from illumina) -m (outputfilename) -s (exclutedreadsfilename)}

\hypertarget{bash-script}{%
\subsection{Bash script}\label{bash-script}}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{\textasciigrave{}}\NormalTok{\#!}\ExtensionTok{/bin/bash}

\KeywordTok{for} \ExtensionTok{i}\NormalTok{ in *.fastq}

\KeywordTok{do}
        \ExtensionTok{sickle}\NormalTok{ pe {-}c }\VariableTok{$i}\NormalTok{ {-}t sanger {-}m }\VariableTok{$i}\NormalTok{.trim.fastq {-}s }\VariableTok{$i}\NormalTok{.singles.fastq}
\KeywordTok{done\textasciigrave{}} 
\end{Highlighting}
\end{Shaded}

\hypertarget{quality-check-with-fastqc}{%
\section{Quality Check with Fastqc}\label{quality-check-with-fastqc}}

\href{https://github.com/s-andrews/FastQC}{Githup}

\hypertarget{about-1}{%
\subsection{About}\label{about-1}}

This will generate HTML files that you can transfer (scp) to your local computer in order to view them in a chrome web page

\hypertarget{exemple-1}{%
\subsection{Exemple}\label{exemple-1}}

Run fastqc on the output (trimmed) file and the non-trimmed file

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{fastqc}\NormalTok{ MF\_combined\_trimmmed.fastq}
\ExtensionTok{fastqc}\NormalTok{ MF\_combined.fastq}
\end{Highlighting}
\end{Shaded}

This will generate two HTML file (one for the combined\_trimmed and one for the combined) that you scp to your local computer in order to view. To scp the file, go to your local computer's terminal and write the following

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{scp}\NormalTok{ karine@146.6.184.205:/home/karine/fastq/MF\_combined\_trimmed\_fastqc.html .}
\FunctionTok{scp}\NormalTok{ karine@146.6.184.205:/home/karine/fastq/MF\_combined\_fastqc.html .}
\end{Highlighting}
\end{Shaded}

Open both files (chrome web page) and compare them (check the quality)

\hypertarget{bash}{%
\subsection{Bash}\label{bash}}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{fastqc}\NormalTok{ *.fastq}
\end{Highlighting}
\end{Shaded}

\hypertarget{fastq-to-fasta-with-seqtk}{%
\section{Fastq to fasta with seqtk}\label{fastq-to-fasta-with-seqtk}}

\href{https://github.com/lh3/seqtk}{Githup}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\tightlist
\item
  Gzip the combined\_trimmed.fastq file
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{gzip}\NormalTok{ MF\_combined\_trimmmed.fastq}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{1}
\tightlist
\item
  Convert the file frome fastq to fasta
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{seqtk}\NormalTok{ seq {-}a MF\_combined\_trimmed.gz }\OperatorTok{>}\NormalTok{ MF.fa}
\end{Highlighting}
\end{Shaded}

\hypertarget{assembly}{%
\chapter{ASSEMBLY}\label{assembly}}

\hypertarget{idba}{%
\section{IDBA}\label{idba}}

\href{https://github.com/loneknightpy/idba}{Github}

\hypertarget{about-2}{%
\subsection{About}\label{about-2}}

The output is a new folder called \texttt{assembly}and the file we are interested in is called \texttt{contig.fa}

\hypertarget{bash-script-1}{%
\subsection{Bash script}\label{bash-script-1}}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{\textasciigrave{}}\NormalTok{\#!}\ExtensionTok{/bin/bash}

\KeywordTok{for} \ExtensionTok{i}\NormalTok{ in *.fa}

\KeywordTok{do}
        \ExtensionTok{idba\_ud}\NormalTok{ {-}l }\VariableTok{$i}\NormalTok{ {-}o }\VariableTok{$i}\NormalTok{.assembly {-}{-}pre\_correction {-}{-}mink 65 {-}{-}maxk 115 {-}{-}step 10 {-}{-}seed\_kmer 55 i{-}{-}num\_threads 20}

\KeywordTok{done}
\end{Highlighting}
\end{Shaded}

\hypertarget{exemple-2}{%
\subsection{Exemple}\label{exemple-2}}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\tightlist
\item
  Create a vi file named \texttt{runidba.sh}and type in the following script. Save and quit (\texttt{ESC}+ \texttt{:qw})
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{idba\_ud}\NormalTok{ {-}l MF.fa {-}o assembly {-}{-}pre\_correction {-}{-}mink 65 {-}{-}maxk 115 {-}{-}step 10 {-}{-}seed\_kmer 55 {-}{-}num\_threads 20}
\end{Highlighting}
\end{Shaded}

\texttt{idba\_ud} \texttt{-l} (imput fasta file) \texttt{-o} (output directory name) \texttt{-\/-pre\_correction} \texttt{-\/-mink} (minimum kmers length) \texttt{-\/-maxk} (maximum kmers length) \texttt{-\/-step} \texttt{-\/-seed\_kmer} \texttt{-\/-num\_threads}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{1}
\tightlist
\item
  Modify the \texttt{runidba.sh} to make it executable
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{chmod}\NormalTok{ +x runidba.sh}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{2}
\tightlist
\item
  Run the script with \texttt{nohup} and \texttt{\&} so it runs in the back and won't be affected by connectivity problem
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{nohup}\NormalTok{ ./runidba.sh }\KeywordTok{\&} 
\end{Highlighting}
\end{Shaded}

\begin{itemize}
\tightlist
\item
  You can view the tasks running on the server by using \texttt{jobs}or \texttt{htop}
\item
  \texttt{free}allows you to see the available memory
\end{itemize}

The output is a new folder called \texttt{assembly}and the file we are interested in is called \texttt{contig.fa}

\hypertarget{megahit}{%
\section{Megahit}\label{megahit}}

\url{https://github.com/voutcn/megahit}

\url{https://www.ncbi.nlm.nih.gov/pubmed/25609793}

MEGAHIT is very fast and memory efficient if time and RAM are causing a bottleneck. Set memory to whatever your machine can handle. This command is set up for interleaved fastq.

In the folder with your fastq files create a vi document called megahit.sh with the following script and make it executable using the chmod +x runmegahit.sh

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#!/bin/bash}

\KeywordTok{for} \ExtensionTok{i}\NormalTok{ in *.fastq}

\KeywordTok{do}

\ExtensionTok{megahit}\NormalTok{ {-}{-}12 }\VariableTok{$i}\NormalTok{ {-}{-}k{-}list 21,33,55,77,99,121 {-}{-}min{-}count 2 {-}{-}verbose {-}t 25 {-}o /home/karine/megahit/}\VariableTok{$i}\NormalTok{ {-}{-}out{-}prefix Megahit\_}\VariableTok{$i}\NormalTok{ {-}{-}memory 80441980}

\KeywordTok{done}
\end{Highlighting}
\end{Shaded}

The FASTA file Megahit\_\$1.contigs.fa is the output

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#!/bin/bash}
\ExtensionTok{megahit}\NormalTok{ {-}{-}12 MF\_combined\_trimmed\_fastq {-}{-}k{-}list 21,33,55,77,99,121 {-}{-}min{-}count 2 {-}{-}verbose {-}t 25 {-}o /home/karine/fastq/megahit {-}{-}out{-}prefix Megahit\_MF {-}{-}memory 80441980}
\end{Highlighting}
\end{Shaded}

By default, the cutoff value is 2, so k-mers occurring at least twice are kept while singleton k-mers are discarded. Because this eliminates not only sequencing errors, but also removes information from genuinely low abundant genome fragments.

\hypertarget{metaspades}{%
\section{metaSPAdes}\label{metaspades}}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{metaspades}\NormalTok{ {-}{-}12 MF\_combined\_trimmed\_fastq {-}o /home/karine/fastq/metaspades {-}m 50}
\end{Highlighting}
\end{Shaded}

\hypertarget{post-assembly-stats}{%
\chapter{POST ASSEMBLY STATS}\label{post-assembly-stats}}

\hypertarget{number-of-contigs}{%
\section{Number of contigs}\label{number-of-contigs}}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{grep}\NormalTok{ {-}c }\StringTok{">"}\NormalTok{ contig.fa }
\end{Highlighting}
\end{Shaded}

\hypertarget{lenght-of-contigs}{%
\section{Lenght of contigs}\label{lenght-of-contigs}}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{seqkit}\NormalTok{ fx2tab {-}{-}length {-}{-}name {-}{-}header{-}line contig.fa}
\end{Highlighting}
\end{Shaded}

\hypertarget{histogram}{%
\section{Histogram}\label{histogram}}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\tightlist
\item
  To obtain a histogram of the lenght of your contigs we first need to extract the column \texttt{lenght} from the document \texttt{contig.fa} with the command \texttt{cut\ -f\ 4\ length.tab} into a new document called \texttt{lenghts}
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cut}\NormalTok{ {-}f 4 length.tab }\OperatorTok{>}\NormalTok{ lengths}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{1}
\item
  Remove the first row of the document lenghts using \texttt{nano}
\item
  Use pipeline to create histogram with the \texttt{lentghs} documents
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{less}\NormalTok{ lenghts }\KeywordTok{|} \ExtensionTok{Rscript}\NormalTok{ {-}e }\StringTok{\textquotesingle{}data=abs(scan(file="stdin")); png("seq.png"); hist(data,xlab="sequences", xlim=c(0,20000))\textquotesingle{}}
\end{Highlighting}
\end{Shaded}

\begin{itemize}
\item
  The output is a png file called ``seq.png''
\item
  If x axis of the histogram is not right change the \texttt{xlim=c(x,x)} values
\end{itemize}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{3}
\tightlist
\item
  Copy the file to your local computer to view it (in your local computer terminal navigate to the local directory where you want the file to be copied)
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{scp}\NormalTok{ karine@146.6.184.205:/home/karine/fastq/assembly/seq.png .}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{4}
\tightlist
\item
  Open document and confirm how many sequences are greater than 2000
\end{enumerate}

\hypertarget{lenght-and-gc}{%
\section{Lenght and GC}\label{lenght-and-gc}}

\hypertarget{run-the-script}{%
\subsection{Run the script}\label{run-the-script}}

High GC organisms tend not to assemble well and may have an uneven read coverage distribution.

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\tightlist
\item
  Run the script length+GC.pl
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{/home/karine/fastq/assembly/length+GC.pl}\NormalTok{ contig.fa }\OperatorTok{>}\NormalTok{ contig\_GC.txt}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{1}
\tightlist
\item
  To view the GC content
\end{enumerate}

\texttt{less\ contig\_GC.txt}

\hypertarget{obtainning-script-from-midgard}{%
\subsection{obtainning script from Midgard}\label{obtainning-script-from-midgard}}

Copy the script from midgard to local computer to asgard

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{scp}\NormalTok{ {-}P 2022 karine@146.6.184.138:/home/scripts/length+GC.pl .}
\FunctionTok{scp}\NormalTok{ length+GC.pl karine@146.6.184.205:/home/karine/fastq/assembly/}
\end{Highlighting}
\end{Shaded}

\hypertarget{keeping-sequence-above-2000kb}{%
\section{Keeping sequence above 2000kb}\label{keeping-sequence-above-2000kb}}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{perl}\NormalTok{ {-}lne }\StringTok{\textquotesingle{}if(/\^{}(>.*)/)\{ $head=$1 \} else \{ $fa\{$head\} .= $\_ \} END\{ foreach $s (keys(\%fa))\{ print "$s\textbackslash{}n$fa\{$s\}\textbackslash{}n" if(length($fa\{$s\})>2000) \}\}\textquotesingle{}}\NormalTok{ contig.fa }\OperatorTok{>}\NormalTok{ 2000kb.fa}
\end{Highlighting}
\end{Shaded}

To count how many you have above 2000 kb

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{grep}\NormalTok{ {-}c }\StringTok{">"}\NormalTok{ 2000kb.fa}
\end{Highlighting}
\end{Shaded}

\hypertarget{binning}{%
\chapter{BINNING}\label{binning}}

Coverage-based binning approaches will require you to map reads to assembled contigs

\hypertarget{mapping}{%
\section{Mapping}\label{mapping}}

\url{http://bio-bwa.sourceforge.net/bwa.shtml}

\hypertarget{ve}{%
\subsection{VE}\label{ve}}

\href{https://github.com/imrambo/genome_mapping}{Github}

This wrapper maps FASTQ reads against an assembly (e.g.~genome) in FASTA format using BWA-MEM.

I put the fasta file with the long sequence as well as the trimmed-combined fastq file of each sample in the same folder, named after the sequence.

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{git}\NormalTok{ clone https://github.com/imrambo/genome\_mapping.git}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  Create a conda environment under your home directory
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{conda}\NormalTok{ env create {-}f environment\_linux64.yml}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\setcounter{enumi}{1}
\tightlist
\item
  Activate the conda environement
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{conda}\NormalTok{ activate scons\_map}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\setcounter{enumi}{2}
\tightlist
\item
  Run a dry run
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{scons}\NormalTok{  {-}{-}dry{-}run {-}{-}fastq\_dir=/home/karine/VP/megahit/concat {-}{-}assembly=/home/karine/VP/megahit/concat/concatenated\_1kb.fa {-}{-}outdir=/home/karine/VP/megahit/concat/map {-}{-}sampleids=fastq\_concat.fastq {-}{-}align\_thread=5 {-}{-}samsort\_thread=5 {-}{-}samsort\_mem=768M {-}{-}nheader=8 {-}{-}tmpdir=/home/karine/tmp {-}{-}logfile=concat.log}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\setcounter{enumi}{3}
\tightlist
\item
  Run the script
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{scons}\NormalTok{  {-}{-}fastq\_dir=/home/karine/VP/SV10 {-}{-}assembly=/home/karine/VP/SV10/SV10\_1kb.fa {-}{-}outdir=/home/karine/VP/SV10\_map {-}{-}sampleids=SV10\_combined {-}{-}align\_thread=5 {-}{-}samsort\_thread=5 {-}{-}samsort\_mem=768M {-}{-}nheader=8 {-}{-}tmpdir=/home/karine/tmp {-}{-}logfile=SV10log}
\end{Highlighting}
\end{Shaded}

\hypertarget{manual}{%
\subsection{manual}\label{manual}}

\textbf{a. Index fasta file using bwa }

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{bwa}\NormalTok{ index 2000kb.fa}
\end{Highlighting}
\end{Shaded}

\textbf{b. Aligne the fasta file against the \texttt{combined\_trimmed.fastq}file}

\begin{itemize}
\tightlist
\item
  Create a \texttt{vi}file named \texttt{mapping.sh}with the following script
\end{itemize}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#!/bin/bash}
\ExtensionTok{bwa}\NormalTok{ mem {-}t 30 2000kb.fa {-}p MF\_combined\_trimmed\_fastq }\OperatorTok{>}\NormalTok{ 2KB.MF.sam}
\end{Highlighting}
\end{Shaded}

\begin{itemize}
\tightlist
\item
  Make the mapping.sh script executable \texttt{chmod\ +x\ mapping.sh}
\item
  Run the script in the background using \texttt{nohup\ ./} and \texttt{\&}
\end{itemize}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{nohup}\NormalTok{ ./mapping.sh }\KeywordTok{\&} 
\end{Highlighting}
\end{Shaded}

\textbf{c.~Convert \texttt{.sam} file to \texttt{.bam} file with} \href{http://www.metagenomics.wiki/tools/samtools}{samtools}

\begin{itemize}
\tightlist
\item
  Create a vi file named \texttt{samtools.sh} with the following script
\end{itemize}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#!/bin/bash}
\ExtensionTok{samtools}\NormalTok{ view {-}b {-}S 2KB.MF.sam }\OperatorTok{>}\NormalTok{ 2KB.MF.bam}
\end{Highlighting}
\end{Shaded}

\texttt{-S}input is a \texttt{.sam} file

\texttt{-b}output is a \texttt{.bam}file

\begin{itemize}
\tightlist
\item
  Make the script executable \texttt{chmod\ +x\ samtools.sh}
\item
  Run the script in the background using \texttt{nohup\ ./} and \texttt{\&}
\end{itemize}

\textbf{d.~Sort the bam file}

Required in order to use the script to generate a depth file

\begin{itemize}
\tightlist
\item
  create a vi file named \texttt{sort.sh} with the following script
\end{itemize}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#!/bin/bash }
\ExtensionTok{samtools}\NormalTok{ sort {-}o 2KB.MF.sorted.bam 2KB.MF.bam}
\end{Highlighting}
\end{Shaded}

\begin{itemize}
\tightlist
\item
  Make the script executable \texttt{chmod\ +x\ sort.sh}
\item
  Run the script in the background using \texttt{nohup\ ./} and \texttt{\&}
\end{itemize}

\hypertarget{depth-file}{%
\section{Depth file}\label{depth-file}}

The depth allows you to know how many sequence you can align with certain sections of your contigs. Section with very little depth (few sequences) are not reputable to use

\hypertarget{run-the-script-1}{%
\subsection{Run the script}\label{run-the-script-1}}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./jgi\_summarize\_bam\_contig\_depths}\NormalTok{ {-}{-}outputDepth depth.txt {-}{-}pairedContigs paired.txt 2KB.MF.sorted.bam}
\end{Highlighting}
\end{Shaded}

To open the depth file \texttt{less\ depth.txt}

\hypertarget{copy-file-from-midgard-to-asgard}{%
\subsection{Copy file from Midgard to Asgard}\label{copy-file-from-midgard-to-asgard}}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{scp}\NormalTok{ {-}P 2022 karine@146.6.184.138:/home/scripts/jgi\_summarize\_bam\_contig\_depths .}
\FunctionTok{scp}\NormalTok{ jgi\_summarize\_bam\_contig\_depths karine@146.6.184.205:/home/karine/fastq/assembly/}
\end{Highlighting}
\end{Shaded}

\hypertarget{metabat}{%
\section{MetaBAT}\label{metabat}}

\href{https://peerj.com/articles/1165/}{MetaBAT}

Efficient tool for accurately reconstructing single genomes from complex microbial communities

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\tightlist
\item
  Create \texttt{vi} file named \texttt{run\_metabat.sh} with the following command
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#!/bin/bash}
\ExtensionTok{metabat2}\NormalTok{ {-}i 2000kb.fa {-}a depth.txt {-}o bins\_dir/bin {-}t 20 {-}{-}minCVSum 0 {-}{-}saveCls {-}d {-}v {-}{-}minCV 0.1 {-}m 2000}
\end{Highlighting}
\end{Shaded}

\texttt{minCVsum} : assigning number of tetranucleotide frequency graphs, don't grab negative numbers
\texttt{-m} : min size of contig to be considered for binning

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{1}
\tightlist
\item
  Run Metabat using \texttt{nohup\ ./}and \texttt{\&}
\end{enumerate}

The output is a folder called \texttt{bins\_dir} containing all the bins created

\hypertarget{bin-quality}{%
\chapter{BIN QUALITY}\label{bin-quality}}

\href{https://github.com/Ecogenomics/CheckM/wiki}{Github}

You have to go back one folder in the terminal (not be in the bins\_dir folder)

you may need to specify the extension of your file for it to work. For example, for file finishing is \texttt{.fa} the command will be \texttt{checkm\ lineage\_wf\ -x\ fa}\ldots{}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{checkm}\NormalTok{ lineage\_wf bins\_dir/ bins\_dir/checkm {-}f bins\_dir/output.txt}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\tightlist
\item
  Open the \texttt{output.txt} document with excel to verify the \textbf{completeness} and \textbf{contamination} of your bins
\end{enumerate}

\begin{itemize}
\tightlist
\item
  Standard : Completeness \textgreater{} 50 \% and Contamination \textless{} 10 \%
\end{itemize}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{1}
\item
  Remove all the spaces with \texttt{control} + \texttt{H}
\item
  Filter the columns by Completeness, and seperate the ones \textless{} 50 \% by adding a line in excel
\item
  Filter by Contamination, and highlight all the ones \textgreater{} 10 \% - These are the bins you want to clean
\end{enumerate}

\hypertarget{bin-cleaning}{%
\chapter{BIN CLEANING}\label{bin-cleaning}}

\href{http://madsalbertsen.github.io/multi-metagenome/docs/overview.html}{MM Genome} uses sequences from two different samples and binning is done by plotting the two coverage estimates against each other.

\href{https://www.nature.com/articles/srep04516}{Vizbin} maps sequences based on tetranucleotide frequency. One can then manually create bins and check the quality of them using CheckM. If you have only one sample (one fastq file) Vizbin should be used.

MM genome uses an R script that will require you to create a virutal environment (VE)

\hypertarget{mm-genome}{%
\section{MM Genome}\label{mm-genome}}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\item
  Create a directory called \texttt{genome\_mapping} under \texttt{/home/username/}
\item
  Clone the Github repository in the genome mapping directory
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{git}\NormalTok{ clone https://github.com/imrambo/genome\_mapping.git}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{2}
\tightlist
\item
  Install miniconda
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{wget}\NormalTok{ https://repo.anaconda.com/miniconda/Miniconda3{-}latest{-}Linux{-}x86\_64.sh}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{bash}\NormalTok{ Miniconda3{-}latest{-}Linux{-}x86\_64.sh}
\end{Highlighting}
\end{Shaded}

Restart your terminal window

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{3}
\tightlist
\item
  See the SCons and local options while in genome\_maping with \texttt{scons\ -h}
\end{enumerate}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{4}
\item
  Create a folfer named bins\_cleaning
\item
  In this folder make a nano text and paste the bin id of the bins you need to clean
\item
  Add .fa at the end of each bin id with the folling command
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{sed} \StringTok{\textquotesingle{}s/$/.fa/g\textquotesingle{}}\NormalTok{ bin\_list.txt}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{7}
\tightlist
\item
  Check to see if everything is in order. If yes use \texttt{-i} to apply the changes
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{sed}\NormalTok{ {-}i }\StringTok{\textquotesingle{}s/$/.fa/g\textquotesingle{}}\NormalTok{ bin\_list.txt}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\roman{enumi}.}
\tightlist
\item
  Go to \texttt{bins\_dir} and use this command to copy the bins that need cleaning in the folder \texttt{bin\_cleaning}
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{for} \ExtensionTok{i}\NormalTok{ in }\KeywordTok{\textasciigrave{}}\FunctionTok{cat}\NormalTok{ bin\_list.txt}\KeywordTok{\textasciigrave{};} \KeywordTok{do} \FunctionTok{cp} \VariableTok{$i}\NormalTok{ ../bin\_cleaning/ }\KeywordTok{;} \KeywordTok{done}
\end{Highlighting}
\end{Shaded}

The Fastq files need to be gzip and in an other directory alone (fastq\_compressed). The compression will take a lot of time so run it in \texttt{screen}

The fastq\_compressed is created in the Fastq directory.

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{gzip}\NormalTok{ {-}c MF\_combined\_trimmed\_fastq }\OperatorTok{>}\NormalTok{ fastq\_compressed/MF\_combined\_trimmed\_fastq.gz }
\end{Highlighting}
\end{Shaded}

\hypertarget{vizbin-1}{%
\section{Vizbin}\label{vizbin-1}}

In your computer, create a folder call Vizbin and in this folder create a folder for each bins that need cleaning and import .fa of that bins from the server to your computer using FileZilla

Open

\hypertarget{depth-file-1}{%
\section{Depth file}\label{depth-file-1}}

In this example we are manually cleaning bin.9.fa

\textbf{a. Copy the bin (bin.9.fa) that you want to clean to a new folder called \texttt{test\_bin\_cleaning}in the fastq directory (you need to be in the \texttt{bins\_directory})}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cp}\NormalTok{ bin.9.fa /home/karine/fastq/test\_bin\_cleaning/}
\end{Highlighting}
\end{Shaded}

\textbf{b. Index the fasta file using bwa}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{bwa}\NormalTok{ index bin.9.fa}
\end{Highlighting}
\end{Shaded}

\textbf{c.~Align the fasta file against the \texttt{combined\_trimmed.fastq} file}

\begin{itemize}
\tightlist
\item
  Create a vi file named \texttt{mapping.sh} with the folling script
\end{itemize}

\begin{Shaded}
\begin{Highlighting}[]
  \CommentTok{\#!/bin/bash}
\ExtensionTok{bwa}\NormalTok{ mem {-}t 30 bin.9.fa {-}p /home/karine/fastq/MF\_combined\_trimmed\_fastq }\OperatorTok{>}\NormalTok{ bin.9.sam }
\end{Highlighting}
\end{Shaded}

\begin{itemize}
\tightlist
\item
  Run the script in the background using \texttt{nohup\ ./} and \texttt{\&}
\end{itemize}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{nohup}\NormalTok{ ./mapping.sh }\KeywordTok{\&} 
\end{Highlighting}
\end{Shaded}

\textbf{d.~Convert \texttt{.sam} file to \texttt{.bam} file with} \href{http://www.metagenomics.wiki/tools/samtools}{samtools}

\emph{Use a vi script}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{samtools}\NormalTok{ view {-}b {-}S bin.9.sam }\OperatorTok{>}\NormalTok{ bin.9.bam}
\end{Highlighting}
\end{Shaded}

\texttt{-S}input is a \texttt{.sam} file

\texttt{-b}output is a \texttt{.bam}file

\textbf{e. Sort the bam file}

Required in order to use the script to generate a depth file

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{samtools}\NormalTok{ sort {-}o bin.9.sorted.bam bin.9.bam}
\end{Highlighting}
\end{Shaded}

\textbf{f.~Create the depth file}

You will need the \texttt{jgi\_summarize\_bam\_contig\_depths} (was located in midgard - SCP it to your folder

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{./jgi\_summarize\_bam\_contig\_depths}\NormalTok{ {-}{-}outputDepth depth.txt {-}{-}pairedContigs paired.txt bin.9.sorted.bam}
\end{Highlighting}
\end{Shaded}

To open the depth file \texttt{less\ depth.txt}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{bin-taxonomy}{%
\chapter{BIN TAXONOMY}\label{bin-taxonomy}}

\hypertarget{barrnap}{%
\section{Barrnap}\label{barrnap}}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\tightlist
\item
  Copy all your cleaned bins into a new foler called cleaned\_bins
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cp}\NormalTok{ *.fa cleaned\_bins/}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{1}
\tightlist
\item
  We want to add the name of the bin to the beginning of every file and change the file type to \texttt{.fna}. Use this perl script :
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{for} \ExtensionTok{i}\NormalTok{ in *.fa }\KeywordTok{;} \KeywordTok{do}  \FunctionTok{perl}\NormalTok{ {-}lne }\StringTok{\textquotesingle{}if(/\^{}>(\textbackslash{}S+)/)\{ print ">$ARGV $1"\} else\{ print \}\textquotesingle{}} \VariableTok{$i} \OperatorTok{>} \VariableTok{$i}\NormalTok{.fna }\KeywordTok{;} \KeywordTok{done}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{2}
\tightlist
\item
  In each of your file, change the space between the name and the scaffhold numer to an underscore
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{sed}\NormalTok{ {-}i }\StringTok{\textquotesingle{}s/ /\_/g\textquotesingle{}}\NormalTok{ *.fna}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{3}
\item
  Move all the \texttt{.fna}to a new directory (\texttt{mkir\ fna})
\item
  Concatenate all the \texttt{.fna} files in one document with the following command
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{ *.fna }\OperatorTok{>}\NormalTok{ all\_bins.fna }
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{4}
\tightlist
\item
  Use barrnap to identify the scaffholds that have partial or complet 16S gene
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{barrnap}\NormalTok{ all\_bins.fna }\OperatorTok{>}\NormalTok{ barrnap\_hits.txt}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{barrnap}\NormalTok{ {-}{-}kingdom arc {-}{-}lencutoff 0.2 {-}{-}reject 0.3 {-}{-}evalue 1e{-}05 all\_bins.fna }\OperatorTok{>}\NormalTok{ barrnap\_archaea.txt}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{barrnap}\NormalTok{ {-}{-}kingdom bac {-}{-}lencutoff 0.2 {-}{-}reject 0.3 {-}{-}evalue 1e{-}05 all\_bins.fna }\OperatorTok{>}\NormalTok{ barrnap\_bacteria.txt}
\end{Highlighting}
\end{Shaded}

Opening the txt file will show you which scaffholds in which contigs have complete or near complete 16S rRNA.

In our case, the bin.40 bin.40\_1.fa\_contig-115\_1168 had the 16S. Therefor, we manually copied the sequence and pasted it in Blast ti identify the organism to which this sequence belonged.

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

To obtain more sequences, we ran barrnap on the inital 2000kb.fa file. We obtained 11 scaffholds containing 16S.

We copied the name of those scaffhold followed by the coordinates of the 16S gene in a vi file named scaff. Example : contig-115\_1168:1-955

The following command was then used to copy the sequence identified in the vi scaff file into a new file name 16S\_2KB.fna

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{for} \ExtensionTok{i}\NormalTok{ in }\KeywordTok{\textasciigrave{}}\FunctionTok{cat}\NormalTok{ scaff}\KeywordTok{\textasciigrave{};} \KeywordTok{do} \ExtensionTok{samtools}\NormalTok{ faidx 2000kb.fa }\VariableTok{$i} \OperatorTok{>>}\NormalTok{16S\_2Kb.fna}\KeywordTok{;} \KeywordTok{done}
\end{Highlighting}
\end{Shaded}

The extracted sequence where then uploaded to Blast and Silva website to identify the bacteria to which those sequence belong.

\hypertarget{gtdbtk}{%
\section{GTDBTK}\label{gtdbtk}}

\href{https://github.com/Ecogenomics/GTDBTk/blob/stable/README.md}{Github}

In the folder with all your clean and completed genomes run this command with nohup

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#!/bin/bash}
\ExtensionTok{gtdbtk}\NormalTok{ classify\_wf {-}{-}cpus 20 {-}{-}genome\_dir /home/karine/Bins/all\_clean {-}{-}out\_dir gtdbk\_output}
\end{Highlighting}
\end{Shaded}

Once it is done running, you can open the folder called \texttt{gtdbk\_output} and copy the folder \texttt{gtdbtk.bac120.summary.tsv} to your local computer in order to open it with excel. Use this folder to identify the phylum, class, order, family and genus that you need to download in order to construct your tree.

\hypertarget{onecodex}{%
\section{OneCodeX}\label{onecodex}}

\hypertarget{phylogenetic-tree}{%
\chapter{PHYLOGENETIC TREE}\label{phylogenetic-tree}}

\hypertarget{download-reference-genomes}{%
\section{Download reference genomes}\label{download-reference-genomes}}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\tightlist
\item
  Download assembly summary genbank file
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{wget}\NormalTok{ ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/assembly\_summary\_genbank.txt}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{1}
\item
  For each phylum, order, class, family of interest, search in the \href{https://www.ncbi.nlm.nih.gov/genome/browse\#!/overview/}{NCBI database} and select 10 individuals with the most complete genome. Copy and paste the name and the assembly number in an excel document. From that excel document, coopy the list of assembly in a vi document called ref.txt
\item
  Search for those identifiers in the assembly summary genbank file with the following command ;
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{for} \ExtensionTok{n}\NormalTok{ in $}\KeywordTok{\textasciigrave{}}\FunctionTok{cat}\NormalTok{ ref.txt}\KeywordTok{\textasciigrave{};} \KeywordTok{do} \FunctionTok{grep} \VariableTok{$n}\NormalTok{ assembly\_summary\_genbank.txt  }\KeywordTok{|} \FunctionTok{cut}\NormalTok{ {-}f 1,7,8,20 }\OperatorTok{>>}\NormalTok{ genomestodownload.tab }\KeywordTok{;} \KeywordTok{done}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{3}
\tightlist
\item
  Modify the paths to download only the fna genomes
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cut}\NormalTok{ {-}f 4 genomestodownload.tab  }\KeywordTok{|}  \FunctionTok{sed} \StringTok{\textquotesingle{}s/$/\textbackslash{}/*fna.gz/g\textquotesingle{}} \OperatorTok{>>}\NormalTok{ ftp.links.txt}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{4}
\tightlist
\item
  Download the genomes and unzip the downloaded files
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{wget}\NormalTok{ {-}i ftp.links.txt}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{gunzip}\NormalTok{ *.gz}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{5}
\tightlist
\item
  Add the file name next to the orginial header
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{for} \ExtensionTok{i}\NormalTok{ in *.fna }\KeywordTok{;} \KeywordTok{do} \FunctionTok{perl}\NormalTok{ {-}lne }\StringTok{\textquotesingle{}if(/\^{}>(\textbackslash{}S+)/)\{ print ">$1 [$ARGV]"\} else\{ print \}\textquotesingle{}} \VariableTok{$i} \OperatorTok{>} \VariableTok{$i}\NormalTok{.MF\_Ref.fna}\KeywordTok{;} \KeywordTok{done} 
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{6}
\item
  Keep only the files that end with \_genomic.fna -\textgreater{} you can move all the others to another folder
\item
  Create a new directory and move your files there
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{mkdir}\NormalTok{ /home/karine/MF\_ref}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{mv}\NormalTok{ *MF\_Ref.fna /home/karine/MF\_ref}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{7}
\tightlist
\item
  Run CheckM
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{checkm}\NormalTok{ lineage\_wf {-}x fna /home/karine/MF\_ref checkm {-}f output\_table.txt}
\end{Highlighting}
\end{Shaded}

Discard the genomes that are not completed.

\hypertarget{phylosift}{%
\section{Phylosift}\label{phylosift}}

\href{https://github.com/gjospin/PhyloSift}{Githup}

Midgard

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\item
  Create a folder containing your completed .fna bins and reference genomes
\item
  Create a vi file named run\_phylosift.sh with the following command and run it using \texttt{nohup\ ./} and \texttt{\&}
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
 \CommentTok{\#!/bin/bash}
\FunctionTok{find}\NormalTok{ . {-}maxdepth 1 {-}name }\StringTok{"*.fna"}\NormalTok{ {-}exec /home/baker/phylosift\_v1.0.1/bin/phylosift search {-}{-}isolate {-}{-}besthit \{\} }\DataTypeTok{\textbackslash{};}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{2}
\tightlist
\item
  Once phylosift is done align the marker genes with the following command using \texttt{nohup\ ./} and \texttt{\&}
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{find}\NormalTok{ . {-}maxdepth 1 {-}name }\StringTok{"*.fna"}\NormalTok{ {-}exec /home/baker/phylosift\_v1.0.1/bin/phylosift align {-}{-}isolate {-}{-}besthit \{\} }\DataTypeTok{\textbackslash{};}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{3}
\tightlist
\item
  Rename and concatenate the aligned marker for both bins
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{find}\NormalTok{ . {-}type f {-}regex }\StringTok{\textquotesingle{}.*alignDir/concat.updated.1.fasta\textquotesingle{}}\NormalTok{   {-}exec cat \{\} }\DataTypeTok{\textbackslash{};} \KeywordTok{|} \FunctionTok{sed}\NormalTok{ {-}r }\StringTok{\textquotesingle{}s/\textbackslash{}.1\textbackslash{}..*//\textquotesingle{}}  \OperatorTok{>}\NormalTok{ name\_of\_bin\_alignmnet.fasta}
\end{Highlighting}
\end{Shaded}

\hypertarget{geneious}{%
\section{Geneious}\label{geneious}}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\item
  Download Geneious software (free 14 days trial) select \texttt{new\ foler} drag and drop the aligned fasta file
\item
  Select \texttt{Align/Assemble} ; \texttt{Multiple\ align\ MUSCLE\ Alignment} \ldots{} ; ok
\item
  Once it is done, remove the bins or sequence that seem too small and re-run \texttt{Multiple\ align\ -\ MUSCLE\ Alignment}
\item
  Once it is done, select the aligned file go to \texttt{Tools} ; \texttt{Mask\ Align}
\item
  Export the file : \texttt{File} ; \texttt{Export} ; \texttt{Selected\ Documents} ; \texttt{Phylip\ alignment\ (*phy)}
\item
  Move the file from your local computer to the server
\end{enumerate}

\hypertarget{raxml}{%
\section{RAxML}\label{raxml}}

\href{https://github.com/stamatak/standard-RAxML}{Github}

In the folder with your .phy file create a vi file named tree.sh with the following command and run it using \texttt{chmod\ +x} ; \texttt{nohup\ ./}and \texttt{\&}

\begin{verbatim}
#!/bin/bash

raxmlHPC-PTHREADS-AVX -T 35 -f a -m PROTGAMMAAUTO -N autoMRE -p 12345 -x 12345 -s MF_aligned.phy -n MF_karine
\end{verbatim}

When the tree is done running, copy the file \texttt{RAxML\_bipartitionsBranchLabels.yourfilename} to your local computer`

\hypertarget{itol}{%
\section{iTOL}\label{itol}}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\item
  Create an \href{https://itol.embl.de/}{iTOL} account and upload your tree (RAxML\_bipartitionsBranchLabels.yourfilename) and visualize the results
\item
  Change the name of all the reference genomes
\end{enumerate}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  Locate the folder where all the reference genoms are located and using terminal view the content of that foler using \texttt{ls\ -l}
\item
  Copy/paste the name into an new tab of the excel document where you have saved the GCA assembly name of the reference genomes.
\item
  Use VLOOKUP to match the GCA assembly name of the reference genomes to the organism name
\item
  In another tab, copy the (1) column with the orginal name of the iTOL tree and the (2) column containing the matching Organism name (in this order very important)
\item
  Copy these two column in a vi file named \texttt{ref\_name\_tree} in the same folder with your tree
\item
  In the same folder create an executable perl script named \texttt{replace\_name.sh} to change the name in your tree
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]

\ExtensionTok{use}\NormalTok{ strict}\KeywordTok{;}
\ExtensionTok{use}\NormalTok{ warnings}\KeywordTok{;}

\ExtensionTok{my} \VariableTok{$treeFile}\NormalTok{ = pop}\KeywordTok{;}
\ExtensionTok{my}\NormalTok{ \%taxonomy = map \{ /(\textbackslash{}S+)\textbackslash{}}\ExtensionTok{s+}\NormalTok{(.+)}\ExtensionTok{/}\NormalTok{; }\VariableTok{$1}\NormalTok{ =}\OperatorTok{>} \VariableTok{$2}\NormalTok{ \} }\OperatorTok{<>}\NormalTok{;}

\ExtensionTok{push}\NormalTok{ @ARGV, }\VariableTok{$treeFile}\KeywordTok{;}

\KeywordTok{while} \KeywordTok{(} \ExtensionTok{my} \VariableTok{$line}\NormalTok{ = }\OperatorTok{<>} \KeywordTok{)} \KeywordTok{\{}
   \VariableTok{$line}\NormalTok{ =}\ExtensionTok{\textasciitilde{}} \ExtensionTok{s}\NormalTok{/\textbackslash{}}\ExtensionTok{b}\VariableTok{$\_}\NormalTok{\textbackslash{}b/}\VariableTok{$taxonomy}\DataTypeTok{\{}\VariableTok{$\_}\DataTypeTok{\}}\NormalTok{/g for keys \%taxonomy}\KeywordTok{;}
   \ExtensionTok{print} \VariableTok{$line}\KeywordTok{;}
\KeywordTok{\}}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\setcounter{enumi}{6}
\tightlist
\item
  Run the perl script
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{perl}\NormalTok{ replace\_name.sh ref\_name\_tree RAxML\_bipartitionsBranchLabels.MF\_karine }\OperatorTok{>}\NormalTok{ new\_tree\_name}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\setcounter{enumi}{7}
\tightlist
\item
  Upload the new created tree to iTOL
\end{enumerate}

\hypertarget{metabolic-pathway}{%
\chapter{METABOLIC PATHWAY}\label{metabolic-pathway}}

\hypertarget{convert-bins-to-protein-.faa}{%
\section{Convert bins to protein (.faa)}\label{convert-bins-to-protein-.faa}}

In the folder with your bins in the .fna format\ldots{}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\tightlist
\item
  Make sure there is no space between the name of you bins and the scaffhold
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{for} \ExtensionTok{i}\NormalTok{ in *.fna }\KeywordTok{;} \KeywordTok{do} \FunctionTok{sed}\NormalTok{ {-}i }\StringTok{\textquotesingle{}s/ /\_/g\textquotesingle{}} \VariableTok{$i} \KeywordTok{;} \KeywordTok{done} 
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{1}
\tightlist
\item
  Run \texttt{prodigal} to convert your bins to amino acid sequence
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{for} \ExtensionTok{i}\NormalTok{ in *.fna }\KeywordTok{;} \KeywordTok{do} \ExtensionTok{prodigal}\NormalTok{ {-}i }\VariableTok{$i}\NormalTok{ {-}o output.txt {-}a }\VariableTok{$i}\NormalTok{.faa }\KeywordTok{;} \KeywordTok{done}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{2}
\item
  Move all the files ending with .faa to another folder \texttt{mkdir\ faa\ ;\ mv\ *.faa\ /home/karine/Bins/all\_clean/fna/faa}
\item
  Remove all the characters after the first space in the header
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{for} \ExtensionTok{i}\NormalTok{ in *.faa}\KeywordTok{;} \KeywordTok{do} \FunctionTok{sed}\NormalTok{ {-}i }\StringTok{\textquotesingle{}s/\textbackslash{}s.*$//\textquotesingle{}} \VariableTok{$i}\KeywordTok{;} \KeywordTok{done} \KeywordTok{\&}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{4}
\item
  Tranfer the files to your local computer
\item
  Submit the files to \href{https://www.genome.jp/tools/kofamkoala/}{kofamKOALA} and \href{https://services.birc.au.dk/hyddb/}{HydDB} - you can only submit one file at a time
\end{enumerate}

\hypertarget{kofamkoala}{%
\subsection{KofamKoala}\label{kofamkoala}}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\item
  Submit one file at a time
\item
  Accept the email
\item
  Save each file as a text file identified with the name of the bin on your computer.
\end{enumerate}

\hypertarget{hyddb}{%
\subsection{HydDB}\label{hyddb}}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\item
  Submit one file at a time and do not close the window while it runs.
\item
  When it's completed, download the excel and with the option text to column seperate the name of the name of bin/contigs and the class prediction.
\item
  Use custom sort to sort by column B (the Hygrogenase group) copy only the groups that are hydrogenase ({[}FeFe{]}, {[}NiFe{]}, {[}fe{]}-)
\item
  Paste these results in an excel document, combining all results from all bins
\end{enumerate}

\hypertarget{hydrogenase-tree}{%
\section{Hydrogenase tree}\label{hydrogenase-tree}}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\tightlist
\item
  In the server (takes a lot of memory) concatenate all your .faa files
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{ *.faa }\OperatorTok{>}\NormalTok{ concat\_faa.faa }
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{1}
\item
  Create a vi document call \texttt{hydDB\_contigs.txt} and copy the name of the sequence from your the excel document containing the name of the contigs identified as hydrogenase
\item
  Extract those sequence from the concatenated file (count after using grep -c ``\textgreater{}'' to make sure the numbers match)
\end{enumerate}

\textbf{Option 1 - La plus rapide}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{pullseq}\NormalTok{ {-}i concat\_faa.faa {-}n hydDB\_contigs.txt}\OperatorTok{>>}\NormalTok{ hydDB\_sequence.faa}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{5}
\tightlist
\item
  View the lenght of the extracted sequences
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{seqkit}\NormalTok{ fx2tab {-}{-}length {-}{-}name {-}{-}header{-}line hydDB\_sequence.faa }\OperatorTok{>>}\NormalTok{ hydDB\_sequence.faa.fasta.lenght}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{6}
\tightlist
\item
  Remove all the smaller sequences (\textless140)
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{perl}\NormalTok{ {-}lne }\StringTok{\textquotesingle{}if(/\^{}(>.*)/)\{ $head=$1 \} else \{ $fa\{$head\} .= $\_ \} END\{ foreach $s (keys(\%fa))\{ print "$s\textbackslash{}n$fa\{$s\}\textbackslash{}n" if(length($fa\{$s\})>140) \}\}\textquotesingle{}}\NormalTok{ hydDB\_sequence.faa }\OperatorTok{>}\NormalTok{ long\_sequence.faa}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{7}
\tightlist
\item
  Count how many you have
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{grep}\NormalTok{ {-}c }\StringTok{">"}\NormalTok{ long\_sequence.faa}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\roman{enumi}.}
\tightlist
\item
  Download the HydrogenaseDataBase fasta file located on the Google Drive
\end{enumerate}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{9}
\tightlist
\item
  Use diamond to compared your sequences with the onces in the DataBase
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{diamond}\NormalTok{ makedb {-}{-}in HydroDB.fasta  {-}d hydroDB}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{10}
\tightlist
\item
  Modify your .faa files to remove any space and turn them into 1 liner
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{for} \ExtensionTok{i}\NormalTok{ in *.faa}\KeywordTok{;} \KeywordTok{do} \FunctionTok{sed}\NormalTok{ {-}i  }\StringTok{\textquotesingle{}s/\textbackslash{}s.*$//\textquotesingle{}} \VariableTok{$i}\KeywordTok{;} \KeywordTok{done} \KeywordTok{\&}  
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{for} \ExtensionTok{i}\NormalTok{ in *.faa }\KeywordTok{;} \KeywordTok{do} \FunctionTok{perl}\NormalTok{ {-}lne }\StringTok{\textquotesingle{}if(/\^{}(>.*)/)\{ $head=$1 \} else \{ $fa\{$head\} .= $\_ \} END\{ foreach $s (sort(keys(\%fa)))\{ print "$s\textbackslash{}n$fa\{$s\}\textbackslash{}n" \}\}\textquotesingle{}} \VariableTok{$i} \OperatorTok{>>} \VariableTok{$i}\NormalTok{.1L.faa }\KeywordTok{;} \KeywordTok{done} 
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{11}
\tightlist
\item
  In the folder with all your .faa bins
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{for} \ExtensionTok{i}\NormalTok{ in *.faa}\KeywordTok{;} \KeywordTok{do} \ExtensionTok{diamond}\NormalTok{ blastp {-}{-}db hydroDB.dmd {-}{-}query }\VariableTok{$i}\NormalTok{ {-}{-}out }\VariableTok{$i}\NormalTok{.hyd.tab {-}{-}threads 3 {-}{-}outfmt 6 qtitle stitle pident length qstart qend sstart send evalue bitscore {-}{-}max{-}target{-}seqs 1 }\KeywordTok{;} \KeywordTok{done}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{12}
\tightlist
\item
  Concat all the \texttt{.hyd.tab} files and open the file with excel. Add these as headers for your document
\end{enumerate}

qtitle \textbar{} stitle \textbar{} pident \textbar{} length \textbar{} Qstart \textbar{} qend \textbar{} Sstart \textbar{} send \textbar{} evalue \textbar{} Bitscore \textbar{}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{13}
\item
  Use VLOOKUP to identify the hydrogenase sequence that were identified throught diamond blast and hydDB
\item
  Keep only the sequence with an alignment length cutoff \textgreater{} 40 amino acid residues and sequence identity \textgreater{} 50\% and Remove Group 4 Hydrogenase hit with sequence identity \textless{} 60\%
\item
  Create a vi document call good\_sequences.txt and extract (pullseq) those sequence from the file with all your sequence
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{pullseq}\NormalTok{ {-}i all\_sequence.faa {-}n good\_sequence.txt}\OperatorTok{>>}\NormalTok{ hydDB\_diamond\_filtered\_sequences.faa}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{16}
\item
  Concat those sequences with the original GoogleDrive data base (I need to open the text document and search for \texttt{bin}, in order to do a manuel \texttt{enter} at the end of the data base sequence and the begining of my sequence, otherwise muscle won't work because there will be \texttt{\textgreater{}} in a sequence)
\item
  Muscle align your sequences in the server (takes a lot of time on the computer)
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{muscle}\NormalTok{ {-}in long\_sequence.faa {-}out aligned\_long\_sequence.faa {-}log log.txt {-}maxiters 2}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\alph{enumi}.}
\setcounter{enumi}{18}
\item
  Transfer files to geneious MAFFT and Mask align
\item
  Eport the file as a Fasta to run fasttree
\item
  Run fasttree command using \texttt{nohup\ ./} and \texttt{\&}
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{\textasciigrave{}}\NormalTok{\#!}\ExtensionTok{/bin/bash}

\ExtensionTok{fasttree}\NormalTok{ hyd\_align\_comp.fasta }\OperatorTok{>}\NormalTok{ hyd.tree}
\end{Highlighting}
\end{Shaded}

\hypertarget{hydrogrenase-database-sequence-lenght}{%
\subsubsection{Hydrogrenase Database sequence lenght}\label{hydrogrenase-database-sequence-lenght}}

To know the lenght of the sequence in the DataBase

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{seqkit}\NormalTok{ fx2tab {-}{-}length {-}{-}name {-}{-}header{-}line hydroDB.faa }\OperatorTok{>>}\NormalTok{ HydroDB\_length.faa}
\end{Highlighting}
\end{Shaded}

Create a histogram

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cut}\NormalTok{ {-}f 4 hydroDB\_length }\OperatorTok{>}\NormalTok{ DBlengths}
\end{Highlighting}
\end{Shaded}

Open DBlengths with vi to remove the first row and run this command to generate the histogram

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{less}\NormalTok{ DBlengths }\KeywordTok{|} \ExtensionTok{Rscript}\NormalTok{ {-}e }\StringTok{\textquotesingle{}data=abs(scan(file="stdin")); png("seq.png"); hist(data,xlab="lengths", xlim=c(0,1000)) \textquotesingle{}}
\end{Highlighting}
\end{Shaded}

\hypertarget{mebs}{%
\section{Mebs}\label{mebs}}

\href{https://github.com/valdeanda/mebs}{github}

git clone \url{https://github.com/valdeanda/mebs.git}

Download mebs in the folder containing all your .faa files

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

Mebs runs on python3 -\textgreater{} download python3 on your computer (used \texttt{brew\ install\ python})

Once python3 is installed create a folder called Mebs containing all your .faa files

Use \texttt{pip\ install} to download these four libraries

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{apt{-}get}\NormalTok{ install python3 python3{-}pip python3{-}matplotlib }\DataTypeTok{\textbackslash{} }\NormalTok{ipython3{-}notebook python3{-}mpltoolkits.basemap}

\ExtensionTok{pip3}\NormalTok{ install {-}U pip}

\ExtensionTok{pip3}\NormalTok{ install {-}{-}upgrade pandas $ sudo {-}H pip3 install {-}{-}upgrade numpy}

\ExtensionTok{pip3}\NormalTok{ install {-}{-}upgrade scipy}

\ExtensionTok{pip3}\NormalTok{ install {-}{-}upgrade seaborn $ sudo {-}H pip3 install {-}U scikit{-}learn}
\end{Highlighting}
\end{Shaded}

Run mebs using perl

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{perl}\NormalTok{ mebs.pl {-}input /home/karine/tree/faa {-}type genomic {-}comp }\OperatorTok{>}\NormalTok{ MF\_karine.tsv}
\end{Highlighting}
\end{Shaded}

In the folder with the output file.tsv run the mebs python script

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{python3}\NormalTok{ mebs\_vis.py Phylo\_5.tsv}
\end{Highlighting}
\end{Shaded}

\hypertarget{metabolic}{%
\section{Metabolic}\label{metabolic}}

\href{https://github.com/AnantharamanLab/METABOLIC}{Githib}

Metabolic was run from the Student folder, with the following command within the folder with the FAA files

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{perl}\NormalTok{ /home/students2020/Tools/METABOLIC/METABOLiC{-}G.pl {-}in{-}gn /home/students2020/karine/folderwithFAA {-}o output/ {-}m /home/students2020/Tools/METABOLIC}
\end{Highlighting}
\end{Shaded}

\hypertarget{other-stuff}{%
\chapter{Other Stuff}\label{other-stuff}}

Create MetabatContig\_list.tsv list needed for DAStool

\hypertarget{one-file}{%
\section{One file}\label{one-file}}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{awk} \StringTok{\textquotesingle{}/>/\{sub(">","\&"2000kb.fa"\_");sub(/\textbackslash{}.fa/,x)\}1\textquotesingle{}}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{grep} \StringTok{">"}\NormalTok{ concat.fasta }\OperatorTok{>}\NormalTok{ concat.txt}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{sed}\NormalTok{ {-}i }\StringTok{\textquotesingle{}s/\textbackslash{}.//g\textquotesingle{}}\NormalTok{ concat.txt}
\FunctionTok{sed}\NormalTok{ {-}i }\StringTok{\textquotesingle{}s/>//g\textquotesingle{}}\NormalTok{ concat.txt}
\end{Highlighting}
\end{Shaded}

\hypertarget{bash-script-2}{%
\section{Bash script}\label{bash-script-2}}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{\textasciigrave{}}\NormalTok{\#!}\ExtensionTok{/bin/bash}

\KeywordTok{for} \ExtensionTok{i}\NormalTok{ in *.faa }

\KeywordTok{do}

\ExtensionTok{VIBRANT\_run.py}\NormalTok{ {-}i }\VariableTok{$i}\NormalTok{ {-}f nucl {-}t 10 {-}l 10000}

\KeywordTok{done} 
\end{Highlighting}
\end{Shaded}

\textbf{Next separate names in concat.txt so have scaffold name followed by sample and bin name example .tsv:}

scaffold\_1326 AB\_69\_metabat\_bin\_100

scaffold\_1711 AB\_69\_metabat\_bin\_100

scaffold\_2201 AB\_69\_metabat\_bin\_100

scaffold\_2419 AB\_69\_metabat\_bin\_10

  \bibliography{book.bib,packages.bib}

\end{document}
