<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Section 6 Texas Protocol | My post-graduate study work</title>
  <meta name="description" content="Section 6 Texas Protocol | My post-graduate study work" />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="Section 6 Texas Protocol | My post-graduate study work" />
  <meta property="og:type" content="book" />
  
  
  
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Section 6 Texas Protocol | My post-graduate study work" />
  
  
  

<meta name="author" content="Karine Villeneuve" />


<meta name="date" content="2020-11-25" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="references.html"/>

<script src="libs/header-attrs-2.3/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">My Research Project</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#about-me"><i class="fa fa-check"></i><b>1.1</b> About me</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="host-bookdown-on-github.html"><a href="host-bookdown-on-github.html"><i class="fa fa-check"></i><b>2</b> Host Bookdown on Github</a>
<ul>
<li class="chapter" data-level="2.1" data-path="host-bookdown-on-github.html"><a href="host-bookdown-on-github.html#getting-bookdown-started"><i class="fa fa-check"></i><b>2.1</b> Getting Bookdown started</a></li>
<li class="chapter" data-level="2.2" data-path="host-bookdown-on-github.html"><a href="host-bookdown-on-github.html#github-repository"><i class="fa fa-check"></i><b>2.2</b> Github Repository</a></li>
<li class="chapter" data-level="2.3" data-path="host-bookdown-on-github.html"><a href="host-bookdown-on-github.html#from-bookdown-to-github"><i class="fa fa-check"></i><b>2.3</b> From Bookdown to Github</a></li>
<li class="chapter" data-level="2.4" data-path="host-bookdown-on-github.html"><a href="host-bookdown-on-github.html#knit-and-upload-to-github"><i class="fa fa-check"></i><b>2.4</b> Knit and upload to github</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="methods.html"><a href="methods.html"><i class="fa fa-check"></i><b>3</b> Methods</a></li>
<li class="chapter" data-level="4" data-path="applications.html"><a href="applications.html"><i class="fa fa-check"></i><b>4</b> Applications</a>
<ul>
<li class="chapter" data-level="4.1" data-path="applications.html"><a href="applications.html#example-one"><i class="fa fa-check"></i><b>4.1</b> Example one</a></li>
<li class="chapter" data-level="4.2" data-path="applications.html"><a href="applications.html#example-two"><i class="fa fa-check"></i><b>4.2</b> Example two</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="final-words.html"><a href="final-words.html"><i class="fa fa-check"></i><b>5</b> Final Words</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="6" data-path="texas-protocol.html"><a href="texas-protocol.html"><i class="fa fa-check"></i><b>6</b> Texas Protocol</a>
<ul>
<li class="chapter" data-level="6.1" data-path="texas-protocol.html"><a href="texas-protocol.html#unzip-the-gz-files"><i class="fa fa-check"></i><b>6.1</b> Unzip the gz files</a></li>
<li class="chapter" data-level="6.2" data-path="texas-protocol.html"><a href="texas-protocol.html#interleaving"><i class="fa fa-check"></i><b>6.2</b> Interleaving</a></li>
<li class="chapter" data-level="6.3" data-path="texas-protocol.html"><a href="texas-protocol.html#sickle"><i class="fa fa-check"></i><b>6.3</b> Sickle</a></li>
<li class="chapter" data-level="6.4" data-path="texas-protocol.html"><a href="texas-protocol.html#fastqc"><i class="fa fa-check"></i><b>6.4</b> Fastqc</a></li>
<li class="chapter" data-level="6.5" data-path="texas-protocol.html"><a href="texas-protocol.html#fastq-to-fasta-with-seqtk"><i class="fa fa-check"></i><b>6.5</b> Fastq to fasta with seqtk</a></li>
<li class="chapter" data-level="6.6" data-path="texas-protocol.html"><a href="texas-protocol.html#assembling"><i class="fa fa-check"></i><b>6.6</b> Assembling</a>
<ul>
<li class="chapter" data-level="6.6.1" data-path="texas-protocol.html"><a href="texas-protocol.html#idba"><i class="fa fa-check"></i><b>6.6.1</b> IDBA</a></li>
<li class="chapter" data-level="6.6.2" data-path="texas-protocol.html"><a href="texas-protocol.html#megahit"><i class="fa fa-check"></i><b>6.6.2</b> Megahit</a></li>
<li class="chapter" data-level="6.6.3" data-path="texas-protocol.html"><a href="texas-protocol.html#metaspades"><i class="fa fa-check"></i><b>6.6.3</b> metaSPAdes</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="texas-protocol.html"><a href="texas-protocol.html#post-assembly-stats"><i class="fa fa-check"></i><b>6.7</b> POST ASSEMBLY STATS</a></li>
<li class="chapter" data-level="6.8" data-path="texas-protocol.html"><a href="texas-protocol.html#lenght-and-gc"><i class="fa fa-check"></i><b>6.8</b> Lenght and GC</a></li>
<li class="chapter" data-level="6.9" data-path="texas-protocol.html"><a href="texas-protocol.html#keeping-sequence-above-2000kb"><i class="fa fa-check"></i><b>6.9</b> Keeping sequence above 2000kb</a></li>
<li class="chapter" data-level="6.10" data-path="texas-protocol.html"><a href="texas-protocol.html#binning"><i class="fa fa-check"></i><b>6.10</b> BINNING</a>
<ul>
<li class="chapter" data-level="6.10.1" data-path="texas-protocol.html"><a href="texas-protocol.html#mapping"><i class="fa fa-check"></i><b>6.10.1</b> Mapping</a></li>
</ul></li>
<li class="chapter" data-level="6.11" data-path="texas-protocol.html"><a href="texas-protocol.html#depth-file"><i class="fa fa-check"></i><b>6.11</b> Depth file</a></li>
<li class="chapter" data-level="6.12" data-path="texas-protocol.html"><a href="texas-protocol.html#metabat"><i class="fa fa-check"></i><b>6.12</b> MetaBAT</a></li>
<li class="chapter" data-level="6.13" data-path="texas-protocol.html"><a href="texas-protocol.html#bin-quality"><i class="fa fa-check"></i><b>6.13</b> BIN QUALITY</a></li>
<li class="chapter" data-level="6.14" data-path="texas-protocol.html"><a href="texas-protocol.html#bin-cleaning"><i class="fa fa-check"></i><b>6.14</b> BIN CLEANING</a>
<ul>
<li class="chapter" data-level="6.14.1" data-path="texas-protocol.html"><a href="texas-protocol.html#mm-genome"><i class="fa fa-check"></i><b>6.14.1</b> MM Genome</a></li>
<li class="chapter" data-level="6.14.2" data-path="texas-protocol.html"><a href="texas-protocol.html#vizbin"><i class="fa fa-check"></i><b>6.14.2</b> Vizbin</a></li>
</ul></li>
<li class="chapter" data-level="6.15" data-path="texas-protocol.html"><a href="texas-protocol.html#depth-file-1"><i class="fa fa-check"></i><b>6.15</b> Depth file</a></li>
<li class="chapter" data-level="6.16" data-path="texas-protocol.html"><a href="texas-protocol.html#bin-taxonomy"><i class="fa fa-check"></i><b>6.16</b> BIN TAXONOMY</a>
<ul>
<li class="chapter" data-level="6.16.1" data-path="texas-protocol.html"><a href="texas-protocol.html#barrnap"><i class="fa fa-check"></i><b>6.16.1</b> Barrnap</a></li>
<li class="chapter" data-level="6.16.2" data-path="texas-protocol.html"><a href="texas-protocol.html#gtdbtk"><i class="fa fa-check"></i><b>6.16.2</b> GTDBTK</a></li>
<li class="chapter" data-level="6.16.3" data-path="texas-protocol.html"><a href="texas-protocol.html#onecodex"><i class="fa fa-check"></i><b>6.16.3</b> OneCodeX</a></li>
</ul></li>
<li class="chapter" data-level="6.17" data-path="texas-protocol.html"><a href="texas-protocol.html#phylogenetic-tree"><i class="fa fa-check"></i><b>6.17</b> PHYLOGENETIC TREE</a>
<ul>
<li class="chapter" data-level="6.17.1" data-path="texas-protocol.html"><a href="texas-protocol.html#phylosift"><i class="fa fa-check"></i><b>6.17.1</b> Phylosift</a></li>
<li class="chapter" data-level="6.17.2" data-path="texas-protocol.html"><a href="texas-protocol.html#geneious"><i class="fa fa-check"></i><b>6.17.2</b> Geneious</a></li>
<li class="chapter" data-level="6.17.3" data-path="texas-protocol.html"><a href="texas-protocol.html#raxml"><i class="fa fa-check"></i><b>6.17.3</b> RAxML</a></li>
</ul></li>
<li class="chapter" data-level="6.18" data-path="texas-protocol.html"><a href="texas-protocol.html#itol"><i class="fa fa-check"></i><b>6.18</b> iTOL</a></li>
<li class="chapter" data-level="6.19" data-path="texas-protocol.html"><a href="texas-protocol.html#metabolic-pathway"><i class="fa fa-check"></i><b>6.19</b> METABOLIC PATHWAY</a></li>
<li class="chapter" data-level="6.20" data-path="texas-protocol.html"><a href="texas-protocol.html#hydrogenase-tree"><i class="fa fa-check"></i><b>6.20</b> Hydrogenase tree</a></li>
<li class="chapter" data-level="6.21" data-path="texas-protocol.html"><a href="texas-protocol.html#mebs"><i class="fa fa-check"></i><b>6.21</b> Mebs</a></li>
<li class="chapter" data-level="6.22" data-path="texas-protocol.html"><a href="texas-protocol.html#metabolic"><i class="fa fa-check"></i><b>6.22</b> Metabolic</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">My post-graduate study work</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="texas-protocol" class="section level1" number="6">
<h1><span class="header-section-number">Section 6</span> Texas Protocol</h1>
<div id="unzip-the-gz-files" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> Unzip the gz files</h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="texas-protocol.html#cb4-1"></a><span class="fu">gunzip</span> *.gz</span></code></pre></div>
<p>Your files should now end in .fastq</p>
</div>
<div id="interleaving" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> Interleaving</h2>
<ol style="list-style-type: lower-alpha">
<li>Run the <font color='green'>interleave_fastq.py</font> script using python</li>
</ol>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="texas-protocol.html#cb5-1"></a><span class="ex">python</span> /home/karine/fastq/interleave_fastq.py MF_R1 MF_R2 <span class="op">&gt;</span> MF_combined.fastq</span></code></pre></div>
<p><em>python /directory/to/script/nameofscript.py file_R1 file_R2 &gt; name_of_combined_file.fastq</em></p>
<ol start="2" style="list-style-type: lower-alpha">
<li>View interleaved file</li>
</ol>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="texas-protocol.html#cb6-1"></a><span class="fu">grep</span> @M MF_combined.fastq <span class="kw">|</span> <span class="fu">head</span></span></code></pre></div>
<p><em>grep (search) <span class="citation">(<span class="citeproc-not-found" data-reference-id="M"><strong>???</strong></span>)</span> (symbol or letter) nameofdocument | (pipeline) head (show me just the first lines)</em></p>
<p>Move all interleaved fastq files into a single folder</p>
</div>
<div id="sickle" class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> Sickle</h2>
<p><a href="https://github.com/najoshi/sickle">Githup</a></p>
<p><strong>About</strong>
Sickle is a tool that uses sliding windows along with quality and length thresholds to determine when quality is sufficiently low to trim the 3’-end of reads and also determines when the quality is sufficiently high enough to trim the 5’-end of reads.</p>
<p><strong>For a single file</strong></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="texas-protocol.html#cb7-1"></a><span class="ex">sickle</span> pe -c MF_combined.fastq -t sanger -m MF_combined_trimmed.fastq -s MF_singles.fastq</span></code></pre></div>
<p><em>sickle pe (paired end) -c (inputfile) -t sanger (from illumina) -m (outputfilename) -s (exclutedreadsfilename)</em></p>
<p><strong>Bash script for multiple files</strong></p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="texas-protocol.html#cb8-1"></a><span class="kw">`</span>#!<span class="ex">/bin/bash</span></span>
<span id="cb8-2"><a href="texas-protocol.html#cb8-2"></a></span>
<span id="cb8-3"><a href="texas-protocol.html#cb8-3"></a><span class="kw">for</span> <span class="ex">i</span> in *.fastq</span>
<span id="cb8-4"><a href="texas-protocol.html#cb8-4"></a></span>
<span id="cb8-5"><a href="texas-protocol.html#cb8-5"></a><span class="kw">do</span></span>
<span id="cb8-6"><a href="texas-protocol.html#cb8-6"></a>        <span class="ex">sickle</span> pe -c <span class="va">$i</span> -t sanger -m <span class="va">$i</span>.trim.fastq -s <span class="va">$i</span>.singles.fastq</span>
<span id="cb8-7"><a href="texas-protocol.html#cb8-7"></a><span class="kw">done`</span> </span></code></pre></div>
</div>
<div id="fastqc" class="section level2" number="6.4">
<h2><span class="header-section-number">6.4</span> Fastqc</h2>
<p><a href="https://github.com/s-andrews/FastQC">Githup</a></p>
<p>This will generate HTML files that you need to transfer to your local computer in order to view them in a chrome web page</p>
<p>Run <font color ='blue'>fastqc</font> on the output (trimmed) file and the non-trimmed file</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="texas-protocol.html#cb9-1"></a><span class="ex">fastqc</span> MF_combined_trimmmed.fastq</span>
<span id="cb9-2"><a href="texas-protocol.html#cb9-2"></a><span class="ex">fastqc</span> MF_combined.fastq</span></code></pre></div>
<ul>
<li><p>This generates two HTML file (one for the combined_trimmed and one for the combined)</p></li>
<li><p>Transfer files to your local computer</p></li>
<li><p>Open both files (chrome web page) and compare them (check the quality)</p></li>
</ul>
<p><strong>Running with Bash for multiple files</strong></p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="texas-protocol.html#cb10-1"></a><span class="ex">fastqc</span> *.fastq</span></code></pre></div>
</div>
<div id="fastq-to-fasta-with-seqtk" class="section level2" number="6.5">
<h2><span class="header-section-number">6.5</span> Fastq to fasta with seqtk</h2>
<p><a href="https://github.com/lh3/seqtk">Githup</a></p>
<ol style="list-style-type: lower-alpha">
<li>Gzip the combined_trimmed.fastq file</li>
</ol>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb11-1"><a href="texas-protocol.html#cb11-1"></a><span class="fu">gzip</span> MF_combined_trimmmed.fastq</span></code></pre></div>
<ol start="2" style="list-style-type: lower-alpha">
<li>Convert the file frome fastq to fasta</li>
</ol>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb12-1"><a href="texas-protocol.html#cb12-1"></a><span class="ex">seqtk</span> seq -a MF_combined_trimmed.gz <span class="op">&gt;</span> MF.fa</span></code></pre></div>
</div>
<div id="assembling" class="section level2" number="6.6">
<h2><span class="header-section-number">6.6</span> Assembling</h2>
<p>There exists multiple assembling programs… it can be hard to chose the best one. I suggest trying a few and seeing what gives you the best results (number of sequence and average length of sequences)</p>
<div id="idba" class="section level3 tabset" number="6.6.1">
<h3><span class="header-section-number">6.6.1</span> IDBA</h3>
<p><a href="https://github.com/loneknightpy/idba">Github</a></p>
<p>Generate a folder called <code>assembly</code>and the file we are interested in is called <code>contig.fa</code></p>
<p><strong>Bash script using VI document</strong></p>
<ol style="list-style-type: decimal">
<li>Create a VI document named <code>runidba.sh</code>containing the following script. To save and quit (<code>ESC</code>+ <code>:qw</code>)</li>
</ol>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb13-1"><a href="texas-protocol.html#cb13-1"></a><span class="kw">`</span>#!<span class="ex">/bin/bash</span></span>
<span id="cb13-2"><a href="texas-protocol.html#cb13-2"></a></span>
<span id="cb13-3"><a href="texas-protocol.html#cb13-3"></a><span class="kw">for</span> <span class="ex">i</span> in *.fa</span>
<span id="cb13-4"><a href="texas-protocol.html#cb13-4"></a></span>
<span id="cb13-5"><a href="texas-protocol.html#cb13-5"></a><span class="kw">do</span></span>
<span id="cb13-6"><a href="texas-protocol.html#cb13-6"></a>        <span class="ex">idba_ud</span> -l <span class="va">$i</span> -o <span class="va">$i</span>.assembly --pre_correction --mink 65 --maxk 115 --step 10 --seed_kmer 55 i--num_threads 20</span>
<span id="cb13-7"><a href="texas-protocol.html#cb13-7"></a></span>
<span id="cb13-8"><a href="texas-protocol.html#cb13-8"></a><span class="kw">done</span></span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Make the <code>runidba.sh</code> to make it executable</li>
</ol>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb14-1"><a href="texas-protocol.html#cb14-1"></a><span class="fu">chmod</span> +x runidba.sh</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Run the script with <code>nohup</code> and <code>&amp;</code> (won’t be affected by connectivity problem)</li>
</ol>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb15-1"><a href="texas-protocol.html#cb15-1"></a><span class="fu">nohup</span> ./runidba.sh <span class="kw">&amp;</span> </span></code></pre></div>
<p><em>View the tasks running on the server by using <code>jobs</code>or <code>htop</code></em></p>
<p><em><code>free</code>allows you to see the available memory</em></p>
<p><strong>For a single file</strong></p>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb16-1"><a href="texas-protocol.html#cb16-1"></a><span class="ex">idba_ud</span> -l MF.fa -o assembly --pre_correction --mink 65 --maxk 115 --step 10 --seed_kmer 55 --num_threads 20</span></code></pre></div>
<p><code>idba_ud</code> <code>-l</code> (imput fasta file) <code>-o</code> (output directory name) <code>--pre_correction</code> <code>--mink</code> (minimum kmers length) <code>--maxk</code> (maximum kmers length) <code>--step</code> <code>--seed_kmer</code> <code>--num_threads</code></p>
</div>
<div id="megahit" class="section level3" number="6.6.2">
<h3><span class="header-section-number">6.6.2</span> Megahit</h3>
<p><a href="https://github.com/voutcn/megahit">Github</a> <a href="https://www.ncbi.nlm.nih.gov/pubmed/25609793">NCBI</a></p>
<p>MEGAHIT is very fast and memory efficient if time and RAM are causing a bottleneck. Set memory to whatever your machine can handle. This command is set up for interleaved fastq.</p>
<ol style="list-style-type: decimal">
<li>Inside the folder containing your <code>fastq</code> files create a VI doc. called <code>megahit.sh</code> with the following script and make it executable</li>
</ol>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb17-1"><a href="texas-protocol.html#cb17-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb17-2"><a href="texas-protocol.html#cb17-2"></a></span>
<span id="cb17-3"><a href="texas-protocol.html#cb17-3"></a><span class="kw">for</span> <span class="ex">i</span> in *.fastq</span>
<span id="cb17-4"><a href="texas-protocol.html#cb17-4"></a></span>
<span id="cb17-5"><a href="texas-protocol.html#cb17-5"></a><span class="kw">do</span></span>
<span id="cb17-6"><a href="texas-protocol.html#cb17-6"></a></span>
<span id="cb17-7"><a href="texas-protocol.html#cb17-7"></a><span class="ex">megahit</span> --12 <span class="va">$i</span> --k-list 21,33,55,77,99,121 --min-count 2 --verbose -t 25 -o /home/karine/megahit/<span class="va">$i</span> --out-prefix Megahit_<span class="va">$i</span> --memory 80441980</span>
<span id="cb17-8"><a href="texas-protocol.html#cb17-8"></a></span>
<span id="cb17-9"><a href="texas-protocol.html#cb17-9"></a><span class="kw">done</span></span></code></pre></div>
<p>The FASTA file Megahit_$1.contigs.fa is the output</p>
<p><strong>For a single file</strong></p>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb18-1"><a href="texas-protocol.html#cb18-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb18-2"><a href="texas-protocol.html#cb18-2"></a><span class="ex">megahit</span> --12 MF_combined_trimmed_fastq --k-list 21,33,55,77,99,121 --min-count 2 --verbose -t 25 -o /home/karine/fastq/megahit --out-prefix Megahit_MF --memory 80441980</span></code></pre></div>
<p>By default, the cutoff value is 2, so k-mers occurring at least twice are kept while singleton k-mers are discarded. Because this eliminates not only sequencing errors, but also removes information from genuinely low abundant genome fragments.</p>
</div>
<div id="metaspades" class="section level3" number="6.6.3">
<h3><span class="header-section-number">6.6.3</span> metaSPAdes</h3>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb19-1"><a href="texas-protocol.html#cb19-1"></a><span class="ex">metaspades</span> --12 MF_combined_trimmed_fastq -o /home/karine/fastq/metaspades -m 50</span></code></pre></div>
</div>
</div>
<div id="post-assembly-stats" class="section level2" number="6.7">
<h2><span class="header-section-number">6.7</span> POST ASSEMBLY STATS</h2>
<p><strong>Number of contigs</strong></p>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb20-1"><a href="texas-protocol.html#cb20-1"></a><span class="fu">grep</span> -c <span class="st">&quot;&gt;&quot;</span> contig.fa </span></code></pre></div>
<p><strong>Lenght of contigs</strong></p>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb21-1"><a href="texas-protocol.html#cb21-1"></a><span class="ex">seqkit</span> fx2tab --length --name --header-line contig.fa</span></code></pre></div>
<p><strong>Histogram</strong></p>
<ol style="list-style-type: lower-alpha">
<li>To obtain a histogram of the lenght of your contigs we first need to extract the column <code>lenght</code> from the document <code>contig.fa</code> with the command <code>cut -f 4 length.tab</code> into a new document called <code>lenghts</code></li>
</ol>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb22-1"><a href="texas-protocol.html#cb22-1"></a><span class="fu">cut</span> -f 4 length.tab <span class="op">&gt;</span> lengths</span></code></pre></div>
<ol start="2" style="list-style-type: lower-alpha">
<li><p>Remove the first row of the document lenghts (open the file in a text editor app like <code>nano</code> or <code>vi</code>)</p></li>
<li><p>Use pipeline to create histogram with the <code>lentghs</code> documents</p></li>
</ol>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb23-1"><a href="texas-protocol.html#cb23-1"></a><span class="fu">less</span> lenghts <span class="kw">|</span> <span class="ex">Rscript</span> -e <span class="st">&#39;data=abs(scan(file=&quot;stdin&quot;)); png(&quot;seq.png&quot;); hist(data,xlab=&quot;sequences&quot;, xlim=c(0,20000))&#39;</span></span></code></pre></div>
<ul>
<li><p>The output is a png file called “seq.png”</p></li>
<li><p>If x axis of the histogram is not right change the <code>xlim=c(x,x)</code> values</p></li>
</ul>
<ol start="4" style="list-style-type: lower-alpha">
<li>Transfer the file to your local computer to view it and confirm how many sequences are greater than 2000</li>
</ol>
</div>
<div id="lenght-and-gc" class="section level2" number="6.8">
<h2><span class="header-section-number">6.8</span> Lenght and GC</h2>
<p>High GC organisms tend not to assemble well and may have an uneven read coverage distribution.</p>
<ol style="list-style-type: decimal">
<li>Run the perl script <font color='green'>length+GC.pl</font></li>
</ol>
<div class="sourceCode" id="cb24"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb24-1"><a href="texas-protocol.html#cb24-1"></a><span class="ex">/home/karine/fastq/assembly/length+GC.pl</span> contig.fa <span class="op">&gt;</span> contig_GC.txt</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>View the GC content <code>less contig_GC.txt</code></li>
</ol>
</div>
<div id="keeping-sequence-above-2000kb" class="section level2" number="6.9">
<h2><span class="header-section-number">6.9</span> Keeping sequence above 2000kb</h2>
<div class="sourceCode" id="cb25"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb25-1"><a href="texas-protocol.html#cb25-1"></a><span class="fu">perl</span> -lne <span class="st">&#39;if(/^(&gt;.*)/){ $head=$1 } else { $fa{$head} .= $_ } END{ foreach $s (keys(%fa)){ print &quot;$s\n$fa{$s}\n&quot; if(length($fa{$s})&gt;2000) }}&#39;</span> contig.fa <span class="op">&gt;</span> 2000kb.fa</span></code></pre></div>
<p>Count how many you have above 2000 kb</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb26-1"><a href="texas-protocol.html#cb26-1"></a><span class="fu">grep</span> -c <span class="st">&quot;&gt;&quot;</span> 2000kb.fa</span></code></pre></div>
</div>
<div id="binning" class="section level2" number="6.10">
<h2><span class="header-section-number">6.10</span> BINNING</h2>
<p>Coverage-based binning approaches will require you to map reads to assembled contigs</p>
<div id="mapping" class="section level3" number="6.10.1">
<h3><span class="header-section-number">6.10.1</span> Mapping</h3>
<p><a href="http://bio-bwa.sourceforge.net/bwa.shtml">BWA Reference Manual</a> and Wrapper <a href="https://github.com/imrambo/genome_mapping">Github</a></p>
<p>This wrapper maps FASTQ reads against an assembly (e.g. genome) in FASTA format using BWA-MEM.</p>
<ol style="list-style-type: decimal">
<li><p>Put the fasta file with the long sequence as well as the trimmed-combined fastq file of each sample in the same folder</p></li>
<li><p>Download the Github repository</p></li>
</ol>
<div class="sourceCode" id="cb27"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb27-1"><a href="texas-protocol.html#cb27-1"></a><span class="fu">git</span> clone https://github.com/imrambo/genome_mapping.git</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Create a conda environment under your home directory</li>
</ol>
<div class="sourceCode" id="cb28"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb28-1"><a href="texas-protocol.html#cb28-1"></a><span class="ex">conda</span> env create -f environment_linux64.yml</span></code></pre></div>
<ol start="4" style="list-style-type: decimal">
<li>Activate the conda environement</li>
</ol>
<div class="sourceCode" id="cb29"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb29-1"><a href="texas-protocol.html#cb29-1"></a><span class="ex">conda</span> activate scons_map</span></code></pre></div>
<ol start="5" style="list-style-type: decimal">
<li>Run a dry run (test run)</li>
</ol>
<div class="sourceCode" id="cb30"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb30-1"><a href="texas-protocol.html#cb30-1"></a><span class="ex">scons</span>  --dry-run --fastq_dir=/home/karine/VP/megahit/concat --assembly=/home/karine/VP/megahit/concat/concatenated_1kb.fa --outdir=/home/karine/VP/megahit/concat/map --sampleids=fastq_concat.fastq --align_thread=5 --samsort_thread=5 --samsort_mem=768M --nheader=8 --tmpdir=/home/karine/tmp --logfile=concat.log</span></code></pre></div>
<ol start="6" style="list-style-type: decimal">
<li>Run the script</li>
</ol>
<div class="sourceCode" id="cb31"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb31-1"><a href="texas-protocol.html#cb31-1"></a><span class="ex">scons</span>  --fastq_dir=/home/karine/VP/SV10 --assembly=/home/karine/VP/SV10/SV10_1kb.fa --outdir=/home/karine/VP/SV10_map --sampleids=SV10_combined --align_thread=5 --samsort_thread=5 --samsort_mem=768M --nheader=8 --tmpdir=/home/karine/tmp --logfile=SV10log</span></code></pre></div>
<p><strong>By hand</strong></p>
<ol style="list-style-type: decimal">
<li>Index fasta file using bwa</li>
</ol>
<div class="sourceCode" id="cb32"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb32-1"><a href="texas-protocol.html#cb32-1"></a><span class="ex">bwa</span> index 2000kb.fa</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Align the fasta file against the <code>combined_trimmed.fastq</code>file</li>
</ol>
<ul>
<li>Create a <code>vi</code>file named <code>mapping.sh</code>with the following script</li>
</ul>
<div class="sourceCode" id="cb33"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb33-1"><a href="texas-protocol.html#cb33-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb33-2"><a href="texas-protocol.html#cb33-2"></a><span class="ex">bwa</span> mem -t 30 2000kb.fa -p MF_combined_trimmed_fastq <span class="op">&gt;</span> 2KB.MF.sam</span></code></pre></div>
<ul>
<li>Make <code>mapping.sh</code> script executable <code>chmod +x mapping.sh</code> and run it script using <code>nohup ./</code> and <code>&amp;</code></li>
</ul>
<ol start="3" style="list-style-type: decimal">
<li>Convert <code>.sam</code> file to <code>.bam</code> file with <a href="http://www.metagenomics.wiki/tools/samtools">samtools</a></li>
</ol>
<ul>
<li>Create a vi file named <code>samtools.sh</code> with the following script</li>
</ul>
<div class="sourceCode" id="cb34"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb34-1"><a href="texas-protocol.html#cb34-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb34-2"><a href="texas-protocol.html#cb34-2"></a><span class="ex">samtools</span> view -b -S 2KB.MF.sam <span class="op">&gt;</span> 2KB.MF.bam</span></code></pre></div>
<p><code>-S</code>input is a <code>.sam</code> file</p>
<p><code>-b</code>output is a <code>.bam</code>file</p>
<ul>
<li>Make script executable <code>chmod +x samtools.sh</code> and run the script in the background using <code>nohup ./</code> and <code>&amp;</code></li>
</ul>
<ol start="4" style="list-style-type: decimal">
<li>Sort the bam file (required in order to use the script to generate a depth file)</li>
</ol>
<ul>
<li>create a vi file named <code>sort.sh</code> with the following script</li>
</ul>
<div class="sourceCode" id="cb35"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb35-1"><a href="texas-protocol.html#cb35-1"></a><span class="co">#!/bin/bash </span></span>
<span id="cb35-2"><a href="texas-protocol.html#cb35-2"></a><span class="ex">samtools</span> sort -o 2KB.MF.sorted.bam 2KB.MF.bam</span></code></pre></div>
<ul>
<li>Make script executable <code>chmod +x sort.sh</code> and run script using <code>nohup ./</code> and <code>&amp;</code></li>
</ul>
</div>
</div>
<div id="depth-file" class="section level2" number="6.11">
<h2><span class="header-section-number">6.11</span> Depth file</h2>
<p>The depth allows you to know how many sequence you can align with certain sections of your contigs. Section with very little depth (few sequences) are not reputable to use</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb36-1"><a href="texas-protocol.html#cb36-1"></a><span class="ex">./jgi_summarize_bam_contig_depths</span> --outputDepth depth.txt --pairedContigs paired.txt 2KB.MF.sorted.bam</span></code></pre></div>
<p>Open ouput depth file with <code>less depth.txt</code></p>
</div>
<div id="metabat" class="section level2" number="6.12">
<h2><span class="header-section-number">6.12</span> MetaBAT</h2>
<p><a href="https://peerj.com/articles/1165/">MetaBAT</a></p>
<p>Efficient tool for accurately reconstructing single genomes from complex microbial communities</p>
<ol style="list-style-type: lower-alpha">
<li>Create <code>vi</code> file named <code>run_metabat.sh</code> with the following command</li>
</ol>
<div class="sourceCode" id="cb37"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb37-1"><a href="texas-protocol.html#cb37-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb37-2"><a href="texas-protocol.html#cb37-2"></a><span class="ex">metabat2</span> -i 2000kb.fa -a depth.txt -o bins_dir/bin -t 20 --minCVSum 0 --saveCls -d -v --minCV 0.1 -m 2000</span></code></pre></div>
<p><code>minCVsum</code> : assigning number of tetranucleotide frequency graphs, don’t grab negative numbers
<code>-m</code> : min size of contig to be considered for binning</p>
<ol start="2" style="list-style-type: lower-alpha">
<li>Run Metabat using <code>nohup ./</code>and <code>&amp;</code></li>
</ol>
<p>The output is a folder called <code>bins_dir</code> containing all the bins created</p>
</div>
<div id="bin-quality" class="section level2" number="6.13">
<h2><span class="header-section-number">6.13</span> BIN QUALITY</h2>
<p><a href="https://github.com/Ecogenomics/CheckM/wiki">Github</a></p>
<p>You have to go back one folder in the terminal (not be in the bins_dir folder)</p>
<p>you may need to specify the extension of your file for it to work. For example, for file finishing is <code>.fa</code> the command will be <code>checkm lineage_wf -x fa</code>…</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb38-1"><a href="texas-protocol.html#cb38-1"></a><span class="ex">checkm</span> lineage_wf bins_dir/ bins_dir/checkm -f bins_dir/output.txt</span></code></pre></div>
<ol style="list-style-type: lower-alpha">
<li>Open the <code>output.txt</code> document with excel to verify the <strong>completeness</strong> and <strong>contamination</strong> of your bins</li>
</ol>
<ul>
<li>Standard : Completeness &gt; 50 % and Contamination &lt; 10 %</li>
</ul>
<ol start="2" style="list-style-type: lower-alpha">
<li><p>Remove all the spaces with <code>control</code> + <code>H</code></p></li>
<li><p>Filter the columns by Completeness, and seperate the ones &lt; 50 % by adding a line in excel</p></li>
<li><p>Filter by Contamination, and highlight all the ones &gt; 10 % - These are the bins you want to clean</p></li>
</ol>
</div>
<div id="bin-cleaning" class="section level2 tabset" number="6.14">
<h2><span class="header-section-number">6.14</span> BIN CLEANING</h2>
<p><a href="http://madsalbertsen.github.io/multi-metagenome/docs/overview.html">MM Genome</a> uses sequences from two different samples and binning is done by plotting the two coverage estimates against each other.</p>
<p><a href="https://www.nature.com/articles/srep04516">Vizbin</a> maps sequences based on tetranucleotide frequency. One can then manually create bins and check the quality of them using CheckM. If you have only one sample (one fastq file) Vizbin should be used.</p>
<p>MM genome uses an R script that will require you to create a virutal environment (VE)</p>
<div id="mm-genome" class="section level3" number="6.14.1">
<h3><span class="header-section-number">6.14.1</span> MM Genome</h3>
<ol style="list-style-type: lower-alpha">
<li><p>Create a directory called <code>genome_mapping</code> under <code>/home/username/</code></p></li>
<li><p>Clone the Github repository in the genome mapping directory</p></li>
</ol>
<div class="sourceCode" id="cb39"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb39-1"><a href="texas-protocol.html#cb39-1"></a><span class="fu">git</span> clone https://github.com/imrambo/genome_mapping.git</span></code></pre></div>
<ol start="3" style="list-style-type: lower-alpha">
<li>Install miniconda</li>
</ol>
<div class="sourceCode" id="cb40"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb40-1"><a href="texas-protocol.html#cb40-1"></a><span class="fu">wget</span> https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh</span></code></pre></div>
<div class="sourceCode" id="cb41"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb41-1"><a href="texas-protocol.html#cb41-1"></a><span class="fu">bash</span> Miniconda3-latest-Linux-x86_64.sh</span></code></pre></div>
<p>Restart your terminal window</p>
<ol start="4" style="list-style-type: lower-alpha">
<li>See the SCons and local options while in genome_maping with <code>scons -h</code></li>
</ol>
<hr />
<ol start="5" style="list-style-type: lower-alpha">
<li><p>Create a folfer named bins_cleaning</p></li>
<li><p>In this folder make a nano text and paste the bin id of the bins you need to clean</p></li>
<li><p>Add .fa at the end of each bin id with the folling command</p></li>
</ol>
<div class="sourceCode" id="cb42"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb42-1"><a href="texas-protocol.html#cb42-1"></a><span class="fu">sed</span> <span class="st">&#39;s/$/.fa/g&#39;</span> bin_list.txt</span></code></pre></div>
<ol start="8" style="list-style-type: lower-alpha">
<li>Check to see if everything is in order. If yes use <code>-i</code> to apply the changes</li>
</ol>
<div class="sourceCode" id="cb43"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb43-1"><a href="texas-protocol.html#cb43-1"></a><span class="fu">sed</span> -i <span class="st">&#39;s/$/.fa/g&#39;</span> bin_list.txt</span></code></pre></div>
<ol style="list-style-type: lower-roman">
<li>Go to <code>bins_dir</code> and use this command to copy the bins that need cleaning in the folder <code>bin_cleaning</code></li>
</ol>
<div class="sourceCode" id="cb44"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb44-1"><a href="texas-protocol.html#cb44-1"></a><span class="kw">for</span> <span class="ex">i</span> in <span class="kw">`</span><span class="fu">cat</span> bin_list.txt<span class="kw">`;</span> <span class="kw">do</span> <span class="fu">cp</span> <span class="va">$i</span> ../bin_cleaning/ <span class="kw">;</span> <span class="kw">done</span></span></code></pre></div>
<p>The Fastq files need to be gzip and in an other directory alone (fastq_compressed). The compression will take a lot of time so run it in <code>screen</code></p>
<p>The fastq_compressed is created in the Fastq directory.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb45-1"><a href="texas-protocol.html#cb45-1"></a><span class="fu">gzip</span> -c MF_combined_trimmed_fastq <span class="op">&gt;</span> fastq_compressed/MF_combined_trimmed_fastq.gz </span></code></pre></div>
</div>
<div id="vizbin" class="section level3" number="6.14.2">
<h3><span class="header-section-number">6.14.2</span> Vizbin</h3>
<p>In your computer, create a folder call Vizbin and in this folder create a folder for each bins that need cleaning and import .fa of that bins from the server to your computer using FileZilla</p>
</div>
</div>
<div id="depth-file-1" class="section level2" number="6.15">
<h2><span class="header-section-number">6.15</span> Depth file</h2>
<p>In this example we are manually cleaning bin.9.fa</p>
<p><strong>a. Copy the bin (bin.9.fa) that you want to clean to a new folder called <code>test_bin_cleaning</code>in the fastq directory (you need to be in the <code>bins_directory</code>)</strong></p>
<div class="sourceCode" id="cb46"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb46-1"><a href="texas-protocol.html#cb46-1"></a><span class="fu">cp</span> bin.9.fa /home/karine/fastq/test_bin_cleaning/</span></code></pre></div>
<p><strong>b. Index the fasta file using bwa</strong></p>
<div class="sourceCode" id="cb47"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb47-1"><a href="texas-protocol.html#cb47-1"></a><span class="ex">bwa</span> index bin.9.fa</span></code></pre></div>
<p><strong>c. Align the fasta file against the <code>combined_trimmed.fastq</code> file</strong></p>
<ul>
<li>Create a vi file named <code>mapping.sh</code> with the folling script</li>
</ul>
<div class="sourceCode" id="cb48"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb48-1"><a href="texas-protocol.html#cb48-1"></a>  <span class="co">#!/bin/bash</span></span>
<span id="cb48-2"><a href="texas-protocol.html#cb48-2"></a><span class="ex">bwa</span> mem -t 30 bin.9.fa -p /home/karine/fastq/MF_combined_trimmed_fastq <span class="op">&gt;</span> bin.9.sam </span></code></pre></div>
<ul>
<li>Run the script in the background using <code>nohup ./</code> and <code>&amp;</code></li>
</ul>
<div class="sourceCode" id="cb49"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb49-1"><a href="texas-protocol.html#cb49-1"></a><span class="fu">nohup</span> ./mapping.sh <span class="kw">&amp;</span> </span></code></pre></div>
<p><strong>d. Convert <code>.sam</code> file to <code>.bam</code> file with</strong> <a href="http://www.metagenomics.wiki/tools/samtools">samtools</a></p>
<p><em>Use a vi script</em></p>
<div class="sourceCode" id="cb50"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb50-1"><a href="texas-protocol.html#cb50-1"></a><span class="ex">samtools</span> view -b -S bin.9.sam <span class="op">&gt;</span> bin.9.bam</span></code></pre></div>
<p><code>-S</code>input is a <code>.sam</code> file</p>
<p><code>-b</code>output is a <code>.bam</code>file</p>
<p><strong>e. Sort the bam file</strong></p>
<p>Required in order to use the script to generate a depth file</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb51-1"><a href="texas-protocol.html#cb51-1"></a><span class="ex">samtools</span> sort -o bin.9.sorted.bam bin.9.bam</span></code></pre></div>
<p><strong>f. Create the depth file</strong></p>
<p>You will need the <code>jgi_summarize_bam_contig_depths</code> (was located in midgard - SCP it to your folder</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb52-1"><a href="texas-protocol.html#cb52-1"></a><span class="ex">./jgi_summarize_bam_contig_depths</span> --outputDepth depth.txt --pairedContigs paired.txt bin.9.sorted.bam</span></code></pre></div>
<p>To open the depth file <code>less depth.txt</code></p>
<hr />
</div>
<div id="bin-taxonomy" class="section level2" number="6.16">
<h2><span class="header-section-number">6.16</span> BIN TAXONOMY</h2>
<div id="barrnap" class="section level3" number="6.16.1">
<h3><span class="header-section-number">6.16.1</span> Barrnap</h3>
<ol style="list-style-type: lower-alpha">
<li>Copy all your cleaned bins into a new foler called cleaned_bins</li>
</ol>
<div class="sourceCode" id="cb53"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb53-1"><a href="texas-protocol.html#cb53-1"></a><span class="fu">cp</span> *.fa cleaned_bins/</span></code></pre></div>
<ol start="2" style="list-style-type: lower-alpha">
<li>We want to add the name of the bin to the beginning of every file and change the file type to <code>.fna</code>. Use this perl script :</li>
</ol>
<div class="sourceCode" id="cb54"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb54-1"><a href="texas-protocol.html#cb54-1"></a><span class="kw">for</span> <span class="ex">i</span> in *.fa <span class="kw">;</span> <span class="kw">do</span>  <span class="fu">perl</span> -lne <span class="st">&#39;if(/^&gt;(\S+)/){ print &quot;&gt;$ARGV $1&quot;} else{ print }&#39;</span> <span class="va">$i</span> <span class="op">&gt;</span> <span class="va">$i</span>.fna <span class="kw">;</span> <span class="kw">done</span></span></code></pre></div>
<ol start="3" style="list-style-type: lower-alpha">
<li>In each of your file, change the space between the name and the scaffhold numer to an underscore</li>
</ol>
<div class="sourceCode" id="cb55"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb55-1"><a href="texas-protocol.html#cb55-1"></a><span class="fu">sed</span> -i <span class="st">&#39;s/ /_/g&#39;</span> *.fna</span></code></pre></div>
<ol start="4" style="list-style-type: lower-alpha">
<li><p>Move all the <code>.fna</code>to a new directory (<code>mkir fna</code>)</p></li>
<li><p>Concatenate all the <code>.fna</code> files in one document with the following command</p></li>
</ol>
<div class="sourceCode" id="cb56"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb56-1"><a href="texas-protocol.html#cb56-1"></a><span class="fu">cat</span> *.fna <span class="op">&gt;</span> all_bins.fna </span></code></pre></div>
<ol start="5" style="list-style-type: lower-alpha">
<li>Use barrnap to identify the scaffholds that have partial or complet 16S gene</li>
</ol>
<div class="sourceCode" id="cb57"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb57-1"><a href="texas-protocol.html#cb57-1"></a><span class="ex">barrnap</span> all_bins.fna <span class="op">&gt;</span> barrnap_hits.txt</span></code></pre></div>
<div class="sourceCode" id="cb58"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb58-1"><a href="texas-protocol.html#cb58-1"></a><span class="ex">barrnap</span> --kingdom arc --lencutoff 0.2 --reject 0.3 --evalue 1e-05 all_bins.fna <span class="op">&gt;</span> barrnap_archaea.txt</span></code></pre></div>
<div class="sourceCode" id="cb59"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb59-1"><a href="texas-protocol.html#cb59-1"></a><span class="ex">barrnap</span> --kingdom bac --lencutoff 0.2 --reject 0.3 --evalue 1e-05 all_bins.fna <span class="op">&gt;</span> barrnap_bacteria.txt</span></code></pre></div>
<p>Opening the txt file will show you which scaffholds in which contigs have complete or near complete 16S rRNA.</p>
<p>In our case, the bin.40 bin.40_1.fa_contig-115_1168 had the 16S. Therefor, we manually copied the sequence and pasted it in Blast ti identify the organism to which this sequence belonged.</p>
<hr />
<p>To obtain more sequences, we ran barrnap on the inital 2000kb.fa file. We obtained 11 scaffholds containing 16S.</p>
<p>We copied the name of those scaffhold followed by the coordinates of the 16S gene in a vi file named scaff. Example : contig-115_1168:1-955</p>
<p>The following command was then used to copy the sequence identified in the vi scaff file into a new file name 16S_2KB.fna</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb60-1"><a href="texas-protocol.html#cb60-1"></a><span class="kw">for</span> <span class="ex">i</span> in <span class="kw">`</span><span class="fu">cat</span> scaff<span class="kw">`;</span> <span class="kw">do</span> <span class="ex">samtools</span> faidx 2000kb.fa <span class="va">$i</span> <span class="op">&gt;&gt;</span>16S_2Kb.fna<span class="kw">;</span> <span class="kw">done</span></span></code></pre></div>
<p>The extracted sequence where then uploaded to Blast and Silva website to identify the bacteria to which those sequence belong.</p>
</div>
<div id="gtdbtk" class="section level3" number="6.16.2">
<h3><span class="header-section-number">6.16.2</span> GTDBTK</h3>
<p><a href="https://github.com/Ecogenomics/GTDBTk/blob/stable/README.md">Github</a></p>
<p>In the folder with all your clean and completed genomes run this command with nohup</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb61-1"><a href="texas-protocol.html#cb61-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb61-2"><a href="texas-protocol.html#cb61-2"></a><span class="ex">gtdbtk</span> classify_wf --cpus 20 --genome_dir /home/karine/Bins/all_clean --out_dir gtdbk_output</span></code></pre></div>
<p>Once it is done running, you can open the folder called <code>gtdbk_output</code> and copy the folder <code>gtdbtk.bac120.summary.tsv</code> to your local computer in order to open it with excel. Use this folder to identify the phylum, class, order, family and genus that you need to download in order to construct your tree.</p>
</div>
<div id="onecodex" class="section level3" number="6.16.3">
<h3><span class="header-section-number">6.16.3</span> OneCodeX</h3>
</div>
</div>
<div id="phylogenetic-tree" class="section level2" number="6.17">
<h2><span class="header-section-number">6.17</span> PHYLOGENETIC TREE</h2>
<p><strong>Download reference genomes</strong></p>
<ol style="list-style-type: lower-alpha">
<li>Download assembly summary genbank file</li>
</ol>
<div class="sourceCode" id="cb62"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb62-1"><a href="texas-protocol.html#cb62-1"></a><span class="fu">wget</span> ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/assembly_summary_genbank.txt</span></code></pre></div>
<ol start="2" style="list-style-type: lower-alpha">
<li><p>For each phylum, order, class, family of interest, search in the <a href="https://www.ncbi.nlm.nih.gov/genome/browse#!/overview/">NCBI database</a> and select 10 individuals with the most complete genome. Copy and paste the name and the assembly number in an excel document. From that excel document, coopy the list of assembly in a vi document called ref.txt</p></li>
<li><p>Search for those identifiers in the assembly summary genbank file with the following command ;</p></li>
</ol>
<div class="sourceCode" id="cb63"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb63-1"><a href="texas-protocol.html#cb63-1"></a><span class="kw">for</span> <span class="ex">n</span> in $<span class="kw">`</span><span class="fu">cat</span> ref.txt<span class="kw">`;</span> <span class="kw">do</span> <span class="fu">grep</span> <span class="va">$n</span> assembly_summary_genbank.txt  <span class="kw">|</span> <span class="fu">cut</span> -f 1,7,8,20 <span class="op">&gt;&gt;</span> genomestodownload.tab <span class="kw">;</span> <span class="kw">done</span></span></code></pre></div>
<ol start="4" style="list-style-type: lower-alpha">
<li>Modify the paths to download only the fna genomes</li>
</ol>
<div class="sourceCode" id="cb64"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb64-1"><a href="texas-protocol.html#cb64-1"></a><span class="fu">cut</span> -f 4 genomestodownload.tab  <span class="kw">|</span>  <span class="fu">sed</span> <span class="st">&#39;s/$/\/*fna.gz/g&#39;</span> <span class="op">&gt;&gt;</span> ftp.links.txt</span></code></pre></div>
<ol start="5" style="list-style-type: lower-alpha">
<li>Download the genomes and unzip the downloaded files</li>
</ol>
<div class="sourceCode" id="cb65"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb65-1"><a href="texas-protocol.html#cb65-1"></a><span class="fu">wget</span> -i ftp.links.txt</span></code></pre></div>
<div class="sourceCode" id="cb66"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb66-1"><a href="texas-protocol.html#cb66-1"></a><span class="fu">gunzip</span> *.gz</span></code></pre></div>
<ol start="6" style="list-style-type: lower-alpha">
<li>Add the file name next to the orginial header</li>
</ol>
<div class="sourceCode" id="cb67"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb67-1"><a href="texas-protocol.html#cb67-1"></a><span class="kw">for</span> <span class="ex">i</span> in *.fna <span class="kw">;</span> <span class="kw">do</span> <span class="fu">perl</span> -lne <span class="st">&#39;if(/^&gt;(\S+)/){ print &quot;&gt;$1 [$ARGV]&quot;} else{ print }&#39;</span> <span class="va">$i</span> <span class="op">&gt;</span> <span class="va">$i</span>.MF_Ref.fna<span class="kw">;</span> <span class="kw">done</span> </span></code></pre></div>
<ol start="7" style="list-style-type: lower-alpha">
<li><p>Keep only the files that end with _genomic.fna -&gt; you can move all the others to another folder</p></li>
<li><p>Create a new directory and move your files there</p></li>
</ol>
<div class="sourceCode" id="cb68"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb68-1"><a href="texas-protocol.html#cb68-1"></a><span class="fu">mkdir</span> /home/karine/MF_ref</span></code></pre></div>
<div class="sourceCode" id="cb69"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb69-1"><a href="texas-protocol.html#cb69-1"></a><span class="fu">mv</span> *MF_Ref.fna /home/karine/MF_ref</span></code></pre></div>
<ol start="8" style="list-style-type: lower-alpha">
<li>Run CheckM</li>
</ol>
<div class="sourceCode" id="cb70"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb70-1"><a href="texas-protocol.html#cb70-1"></a><span class="ex">checkm</span> lineage_wf -x fna /home/karine/MF_ref checkm -f output_table.txt</span></code></pre></div>
<p>Discard the genomes that are not completed.</p>
<div id="phylosift" class="section level3" number="6.17.1">
<h3><span class="header-section-number">6.17.1</span> Phylosift</h3>
<p><a href="https://github.com/gjospin/PhyloSift">Githup</a></p>
<p><font color ='red'>Midgard</font></p>
<ol style="list-style-type: lower-alpha">
<li><p>Create a folder containing your completed .fna bins and reference genomes</p></li>
<li><p>Create a vi file named run_phylosift.sh with the following command and run it using <code>nohup ./</code> and <code>&amp;</code></p></li>
</ol>
<div class="sourceCode" id="cb71"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb71-1"><a href="texas-protocol.html#cb71-1"></a> <span class="co">#!/bin/bash</span></span>
<span id="cb71-2"><a href="texas-protocol.html#cb71-2"></a><span class="fu">find</span> . -maxdepth 1 -name <span class="st">&quot;*.fna&quot;</span> -exec /home/baker/phylosift_v1.0.1/bin/phylosift search --isolate --besthit {} <span class="dt">\;</span></span></code></pre></div>
<ol start="3" style="list-style-type: lower-alpha">
<li>Once phylosift is done align the marker genes with the following command using <code>nohup ./</code> and <code>&amp;</code></li>
</ol>
<div class="sourceCode" id="cb72"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb72-1"><a href="texas-protocol.html#cb72-1"></a><span class="fu">find</span> . -maxdepth 1 -name <span class="st">&quot;*.fna&quot;</span> -exec /home/baker/phylosift_v1.0.1/bin/phylosift align --isolate --besthit {} <span class="dt">\;</span></span></code></pre></div>
<ol start="4" style="list-style-type: lower-alpha">
<li>Rename and concatenate the aligned marker for both bins</li>
</ol>
<div class="sourceCode" id="cb73"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb73-1"><a href="texas-protocol.html#cb73-1"></a><span class="fu">find</span> . -type f -regex <span class="st">&#39;.*alignDir/concat.updated.1.fasta&#39;</span>   -exec cat {} <span class="dt">\;</span> <span class="kw">|</span> <span class="fu">sed</span> -r <span class="st">&#39;s/\.1\..*//&#39;</span>  <span class="op">&gt;</span> name_of_bin_alignmnet.fasta</span></code></pre></div>
</div>
<div id="geneious" class="section level3" number="6.17.2">
<h3><span class="header-section-number">6.17.2</span> Geneious</h3>
<ol style="list-style-type: lower-alpha">
<li><p>Download Geneious software (free 14 days trial) select <code>new foler</code> drag and drop the aligned fasta file</p></li>
<li><p>Select <code>Align/Assemble</code> ; <code>Multiple align MUSCLE Alignment</code> … ; ok</p></li>
<li><p>Once it is done, remove the bins or sequence that seem too small and re-run <code>Multiple align - MUSCLE Alignment</code></p></li>
<li><p>Once it is done, select the aligned file go to <code>Tools</code> ; <code>Mask Align</code></p></li>
<li><p>Export the file : <code>File</code> ; <code>Export</code> ; <code>Selected Documents</code> ; <code>Phylip alignment (*phy)</code></p></li>
<li><p>Move the file from your local computer to the server</p></li>
</ol>
</div>
<div id="raxml" class="section level3" number="6.17.3">
<h3><span class="header-section-number">6.17.3</span> RAxML</h3>
<p><a href="https://github.com/stamatak/standard-RAxML">Github</a></p>
<p>In the folder with your .phy file create a vi file named tree.sh with the following command and run it using <code>chmod +x</code> ; <code>nohup ./</code>and <code>&amp;</code></p>
<pre><code>#!/bin/bash

raxmlHPC-PTHREADS-AVX -T 35 -f a -m PROTGAMMAAUTO -N autoMRE -p 12345 -x 12345 -s MF_aligned.phy -n MF_karine</code></pre>
<p>When the tree is done running, copy the file <code>RAxML_bipartitionsBranchLabels.yourfilename</code> to your local computer`</p>
</div>
</div>
<div id="itol" class="section level2" number="6.18">
<h2><span class="header-section-number">6.18</span> iTOL</h2>
<ol style="list-style-type: lower-alpha">
<li><p>Create an <a href="https://itol.embl.de/">iTOL</a> account and upload your tree (RAxML_bipartitionsBranchLabels.yourfilename) and visualize the results</p></li>
<li><p>Change the name of all the reference genomes</p></li>
</ol>
<ol style="list-style-type: decimal">
<li>Locate the folder where all the reference genoms are located and using terminal view the content of that foler using <code>ls -l</code></li>
<li>Copy/paste the name into an new tab of the excel document where you have saved the GCA assembly name of the reference genomes.</li>
<li>Use VLOOKUP to match the GCA assembly name of the reference genomes to the organism name</li>
<li>In another tab, copy the (1) column with the orginal name of the iTOL tree and the (2) column containing the matching Organism name (in this order very important)</li>
<li>Copy these two column in a vi file named <code>ref_name_tree</code> in the same folder with your tree</li>
<li>In the same folder create an executable perl script named <code>replace_name.sh</code> to change the name in your tree</li>
</ol>
<div class="sourceCode" id="cb75"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb75-1"><a href="texas-protocol.html#cb75-1"></a></span>
<span id="cb75-2"><a href="texas-protocol.html#cb75-2"></a><span class="ex">use</span> strict<span class="kw">;</span></span>
<span id="cb75-3"><a href="texas-protocol.html#cb75-3"></a><span class="ex">use</span> warnings<span class="kw">;</span></span>
<span id="cb75-4"><a href="texas-protocol.html#cb75-4"></a></span>
<span id="cb75-5"><a href="texas-protocol.html#cb75-5"></a><span class="ex">my</span> <span class="va">$treeFile</span> = pop<span class="kw">;</span></span>
<span id="cb75-6"><a href="texas-protocol.html#cb75-6"></a><span class="ex">my</span> %taxonomy = map { /(\S+)\<span class="ex">s+</span>(.+)<span class="ex">/</span>; <span class="va">$1</span> =<span class="op">&gt;</span> <span class="va">$2</span> } <span class="op">&lt;&gt;</span>;</span>
<span id="cb75-7"><a href="texas-protocol.html#cb75-7"></a></span>
<span id="cb75-8"><a href="texas-protocol.html#cb75-8"></a><span class="ex">push</span> @ARGV, <span class="va">$treeFile</span><span class="kw">;</span></span>
<span id="cb75-9"><a href="texas-protocol.html#cb75-9"></a></span>
<span id="cb75-10"><a href="texas-protocol.html#cb75-10"></a><span class="kw">while</span> <span class="kw">(</span> <span class="ex">my</span> <span class="va">$line</span> = <span class="op">&lt;&gt;</span> <span class="kw">)</span> <span class="kw">{</span></span>
<span id="cb75-11"><a href="texas-protocol.html#cb75-11"></a>   <span class="va">$line</span> =<span class="ex">~</span> <span class="ex">s</span>/\<span class="ex">b</span><span class="va">$_</span>\b/<span class="va">$taxonomy</span><span class="dt">{</span><span class="va">$_</span><span class="dt">}</span>/g for keys %taxonomy<span class="kw">;</span></span>
<span id="cb75-12"><a href="texas-protocol.html#cb75-12"></a>   <span class="ex">print</span> <span class="va">$line</span><span class="kw">;</span></span>
<span id="cb75-13"><a href="texas-protocol.html#cb75-13"></a><span class="kw">}</span></span></code></pre></div>
<ol start="7" style="list-style-type: decimal">
<li>Run the perl script</li>
</ol>
<div class="sourceCode" id="cb76"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb76-1"><a href="texas-protocol.html#cb76-1"></a><span class="fu">perl</span> replace_name.sh ref_name_tree RAxML_bipartitionsBranchLabels.MF_karine <span class="op">&gt;</span> new_tree_name</span></code></pre></div>
<ol start="8" style="list-style-type: decimal">
<li>Upload the new created tree to iTOL</li>
</ol>
</div>
<div id="metabolic-pathway" class="section level2" number="6.19">
<h2><span class="header-section-number">6.19</span> METABOLIC PATHWAY</h2>
<p><strong>Convert bins to protein (.faa)</strong></p>
<p>In the folder with your bins in the .fna format…</p>
<ol style="list-style-type: lower-alpha">
<li>Make sure there is no space between the name of you bins and the scaffhold</li>
</ol>
<div class="sourceCode" id="cb77"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb77-1"><a href="texas-protocol.html#cb77-1"></a><span class="kw">for</span> <span class="ex">i</span> in *.fna <span class="kw">;</span> <span class="kw">do</span> <span class="fu">sed</span> -i <span class="st">&#39;s/ /_/g&#39;</span> <span class="va">$i</span> <span class="kw">;</span> <span class="kw">done</span> </span></code></pre></div>
<ol start="2" style="list-style-type: lower-alpha">
<li>Run <code>prodigal</code> to convert your bins to amino acid sequence</li>
</ol>
<div class="sourceCode" id="cb78"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb78-1"><a href="texas-protocol.html#cb78-1"></a><span class="kw">for</span> <span class="ex">i</span> in *.fna <span class="kw">;</span> <span class="kw">do</span> <span class="ex">prodigal</span> -i <span class="va">$i</span> -o output.txt -a <span class="va">$i</span>.faa <span class="kw">;</span> <span class="kw">done</span></span></code></pre></div>
<ol start="3" style="list-style-type: lower-alpha">
<li><p>Move all the files ending with .faa to another folder <code>mkdir faa ; mv *.faa /home/karine/Bins/all_clean/fna/faa</code></p></li>
<li><p>Remove all the characters after the first space in the header</p></li>
</ol>
<div class="sourceCode" id="cb79"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb79-1"><a href="texas-protocol.html#cb79-1"></a><span class="kw">for</span> <span class="ex">i</span> in *.faa<span class="kw">;</span> <span class="kw">do</span> <span class="fu">sed</span> -i <span class="st">&#39;s/\s.*$//&#39;</span> <span class="va">$i</span><span class="kw">;</span> <span class="kw">done</span> <span class="kw">&amp;</span></span></code></pre></div>
<ol start="5" style="list-style-type: lower-alpha">
<li><p>Tranfer the files to your local computer</p></li>
<li><p>Submit the files to <a href="https://www.genome.jp/tools/kofamkoala/">kofamKOALA</a> and <a href="https://services.birc.au.dk/hyddb/">HydDB</a> - you can only submit one file at a time</p></li>
</ol>
<p><strong>KofamKoala</strong></p>
<ol style="list-style-type: lower-alpha">
<li><p>Submit one file at a time</p></li>
<li><p>Accept the email</p></li>
<li><p>Save each file as a text file identified with the name of the bin on your computer.</p></li>
</ol>
<p><strong>HydDB</strong></p>
<ol style="list-style-type: lower-alpha">
<li><p>Submit one file at a time and do not close the window while it runs.</p></li>
<li><p>When it’s completed, download the excel and with the option text to column seperate the name of the name of bin/contigs and the class prediction.</p></li>
<li><p>Use custom sort to sort by column B (the Hygrogenase group) copy only the groups that are hydrogenase ([FeFe], [NiFe], [fe]-)</p></li>
<li><p>Paste these results in an excel document, combining all results from all bins</p></li>
</ol>
</div>
<div id="hydrogenase-tree" class="section level2" number="6.20">
<h2><span class="header-section-number">6.20</span> Hydrogenase tree</h2>
<ol style="list-style-type: lower-alpha">
<li>In the server (takes a lot of memory) concatenate all your .faa files</li>
</ol>
<div class="sourceCode" id="cb80"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb80-1"><a href="texas-protocol.html#cb80-1"></a><span class="fu">cat</span> *.faa <span class="op">&gt;</span> concat_faa.faa </span></code></pre></div>
<ol start="2" style="list-style-type: lower-alpha">
<li><p>Create a vi document call <code>hydDB_contigs.txt</code> and copy the name of the sequence from your the excel document containing the name of the contigs identified as hydrogenase</p></li>
<li><p>Extract those sequence from the concatenated file (count after using grep -c “&gt;” to make sure the numbers match)</p></li>
</ol>
<p><strong>Option 1 - La plus rapide</strong></p>
<div class="sourceCode" id="cb81"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb81-1"><a href="texas-protocol.html#cb81-1"></a><span class="ex">pullseq</span> -i concat_faa.faa -n hydDB_contigs.txt<span class="op">&gt;&gt;</span> hydDB_sequence.faa</span></code></pre></div>
<ol start="6" style="list-style-type: lower-alpha">
<li>View the lenght of the extracted sequences</li>
</ol>
<div class="sourceCode" id="cb82"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb82-1"><a href="texas-protocol.html#cb82-1"></a><span class="ex">seqkit</span> fx2tab --length --name --header-line hydDB_sequence.faa <span class="op">&gt;&gt;</span> hydDB_sequence.faa.fasta.lenght</span></code></pre></div>
<ol start="7" style="list-style-type: lower-alpha">
<li>Remove all the smaller sequences (&lt;140)</li>
</ol>
<div class="sourceCode" id="cb83"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb83-1"><a href="texas-protocol.html#cb83-1"></a><span class="fu">perl</span> -lne <span class="st">&#39;if(/^(&gt;.*)/){ $head=$1 } else { $fa{$head} .= $_ } END{ foreach $s (keys(%fa)){ print &quot;$s\n$fa{$s}\n&quot; if(length($fa{$s})&gt;140) }}&#39;</span> hydDB_sequence.faa <span class="op">&gt;</span> long_sequence.faa</span></code></pre></div>
<ol start="8" style="list-style-type: lower-alpha">
<li>Count how many you have</li>
</ol>
<div class="sourceCode" id="cb84"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb84-1"><a href="texas-protocol.html#cb84-1"></a><span class="fu">grep</span> -c <span class="st">&quot;&gt;&quot;</span> long_sequence.faa</span></code></pre></div>
<ol style="list-style-type: lower-roman">
<li>Download the HydrogenaseDataBase fasta file located on the Google Drive</li>
</ol>
<ol start="10" style="list-style-type: lower-alpha">
<li>Use diamond to compared your sequences with the onces in the DataBase</li>
</ol>
<div class="sourceCode" id="cb85"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb85-1"><a href="texas-protocol.html#cb85-1"></a><span class="ex">diamond</span> makedb --in HydroDB.fasta  -d hydroDB</span></code></pre></div>
<ol start="11" style="list-style-type: lower-alpha">
<li>Modify your .faa files to remove any space and turn them into 1 liner</li>
</ol>
<div class="sourceCode" id="cb86"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb86-1"><a href="texas-protocol.html#cb86-1"></a><span class="kw">for</span> <span class="ex">i</span> in *.faa<span class="kw">;</span> <span class="kw">do</span> <span class="fu">sed</span> -i  <span class="st">&#39;s/\s.*$//&#39;</span> <span class="va">$i</span><span class="kw">;</span> <span class="kw">done</span> <span class="kw">&amp;</span>  </span></code></pre></div>
<div class="sourceCode" id="cb87"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb87-1"><a href="texas-protocol.html#cb87-1"></a><span class="kw">for</span> <span class="ex">i</span> in *.faa <span class="kw">;</span> <span class="kw">do</span> <span class="fu">perl</span> -lne <span class="st">&#39;if(/^(&gt;.*)/){ $head=$1 } else { $fa{$head} .= $_ } END{ foreach $s (sort(keys(%fa))){ print &quot;$s\n$fa{$s}\n&quot; }}&#39;</span> <span class="va">$i</span> <span class="op">&gt;&gt;</span> <span class="va">$i</span>.1L.faa <span class="kw">;</span> <span class="kw">done</span> </span></code></pre></div>
<ol start="12" style="list-style-type: lower-alpha">
<li>In the folder with all your .faa bins</li>
</ol>
<div class="sourceCode" id="cb88"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb88-1"><a href="texas-protocol.html#cb88-1"></a><span class="kw">for</span> <span class="ex">i</span> in *.faa<span class="kw">;</span> <span class="kw">do</span> <span class="ex">diamond</span> blastp --db hydroDB.dmd --query <span class="va">$i</span> --out <span class="va">$i</span>.hyd.tab --threads 3 --outfmt 6 qtitle stitle pident length qstart qend sstart send evalue bitscore --max-target-seqs 1 <span class="kw">;</span> <span class="kw">done</span></span></code></pre></div>
<ol start="13" style="list-style-type: lower-alpha">
<li>Concat all the <code>.hyd.tab</code> files and open the file with excel. Add these as headers for your document</li>
</ol>
<div class="line-block">qtitle | stitle | pident | length | Qstart | qend | Sstart | send | evalue | Bitscore |</div>
<ol start="14" style="list-style-type: lower-alpha">
<li><p>Use VLOOKUP to identify the hydrogenase sequence that were identified throught diamond blast and hydDB</p></li>
<li><p>Keep only the sequence with an alignment length cutoff &gt; 40 amino acid residues and sequence identity &gt; 50% and Remove Group 4 Hydrogenase hit with sequence identity &lt; 60%</p></li>
<li><p>Create a vi document call good_sequences.txt and extract (pullseq) those sequence from the file with all your sequence</p></li>
</ol>
<div class="sourceCode" id="cb89"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb89-1"><a href="texas-protocol.html#cb89-1"></a><span class="ex">pullseq</span> -i all_sequence.faa -n good_sequence.txt<span class="op">&gt;&gt;</span> hydDB_diamond_filtered_sequences.faa</span></code></pre></div>
<ol start="17" style="list-style-type: lower-alpha">
<li><p>Concat those sequences with the original GoogleDrive data base (I need to open the text document and search for <code>bin</code>, in order to do a manuel <code>enter</code> at the end of the data base sequence and the begining of my sequence, otherwise muscle won’t work because there will be <code>&gt;</code> in a sequence)</p></li>
<li><p>Muscle align your sequences in the server (takes a lot of time on the computer)</p></li>
</ol>
<div class="sourceCode" id="cb90"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb90-1"><a href="texas-protocol.html#cb90-1"></a><span class="ex">muscle</span> -in long_sequence.faa -out aligned_long_sequence.faa -log log.txt -maxiters 2</span></code></pre></div>
<ol start="19" style="list-style-type: lower-alpha">
<li><p>Transfer files to geneious MAFFT and Mask align</p></li>
<li><p>Eport the file as a Fasta to run fasttree</p></li>
<li><p>Run fasttree command using <code>nohup ./</code> and <code>&amp;</code></p></li>
</ol>
<div class="sourceCode" id="cb91"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb91-1"><a href="texas-protocol.html#cb91-1"></a><span class="kw">`</span>#!<span class="ex">/bin/bash</span></span>
<span id="cb91-2"><a href="texas-protocol.html#cb91-2"></a></span>
<span id="cb91-3"><a href="texas-protocol.html#cb91-3"></a><span class="ex">fasttree</span> hyd_align_comp.fasta <span class="op">&gt;</span> hyd.tree</span></code></pre></div>
<p><strong>Hydrogrenase Database sequence lenght</strong></p>
<p>To know the lenght of the sequence in the DataBase</p>
<div class="sourceCode" id="cb92"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb92-1"><a href="texas-protocol.html#cb92-1"></a><span class="ex">seqkit</span> fx2tab --length --name --header-line hydroDB.faa <span class="op">&gt;&gt;</span> HydroDB_length.faa</span></code></pre></div>
<p>Create a histogram</p>
<div class="sourceCode" id="cb93"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb93-1"><a href="texas-protocol.html#cb93-1"></a><span class="fu">cut</span> -f 4 hydroDB_length <span class="op">&gt;</span> DBlengths</span></code></pre></div>
<p>Open DBlengths with vi to remove the first row and run this command to generate the histogram</p>
<div class="sourceCode" id="cb94"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb94-1"><a href="texas-protocol.html#cb94-1"></a><span class="fu">less</span> DBlengths <span class="kw">|</span> <span class="ex">Rscript</span> -e <span class="st">&#39;data=abs(scan(file=&quot;stdin&quot;)); png(&quot;seq.png&quot;); hist(data,xlab=&quot;lengths&quot;, xlim=c(0,1000)) &#39;</span></span></code></pre></div>
</div>
<div id="mebs" class="section level2" number="6.21">
<h2><span class="header-section-number">6.21</span> Mebs</h2>
<p><a href="https://github.com/valdeanda/mebs">github</a></p>
<p>git clone <a href="https://github.com/valdeanda/mebs.git" class="uri">https://github.com/valdeanda/mebs.git</a></p>
<p>Download mebs in the folder containing all your .faa files</p>
<hr />
<p>Mebs runs on python3 -&gt; download python3 on your computer (used <code>brew install python</code>)</p>
<p>Once python3 is installed create a folder called Mebs containing all your .faa files</p>
<p>Use <code>pip install</code> to download these four libraries</p>
<div class="sourceCode" id="cb95"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb95-1"><a href="texas-protocol.html#cb95-1"></a><span class="ex">apt-get</span> install python3 python3-pip python3-matplotlib <span class="dt">\ </span>ipython3-notebook python3-mpltoolkits.basemap</span>
<span id="cb95-2"><a href="texas-protocol.html#cb95-2"></a></span>
<span id="cb95-3"><a href="texas-protocol.html#cb95-3"></a><span class="ex">pip3</span> install -U pip</span>
<span id="cb95-4"><a href="texas-protocol.html#cb95-4"></a></span>
<span id="cb95-5"><a href="texas-protocol.html#cb95-5"></a><span class="ex">pip3</span> install --upgrade pandas $ sudo -H pip3 install --upgrade numpy</span>
<span id="cb95-6"><a href="texas-protocol.html#cb95-6"></a></span>
<span id="cb95-7"><a href="texas-protocol.html#cb95-7"></a><span class="ex">pip3</span> install --upgrade scipy</span>
<span id="cb95-8"><a href="texas-protocol.html#cb95-8"></a></span>
<span id="cb95-9"><a href="texas-protocol.html#cb95-9"></a><span class="ex">pip3</span> install --upgrade seaborn $ sudo -H pip3 install -U scikit-learn</span></code></pre></div>
<p>Run mebs using perl</p>
<div class="sourceCode" id="cb96"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb96-1"><a href="texas-protocol.html#cb96-1"></a><span class="fu">perl</span> mebs.pl -input /home/karine/tree/faa -type genomic -comp <span class="op">&gt;</span> MF_karine.tsv</span></code></pre></div>
<p>In the folder with the output file.tsv run the mebs python script</p>
<div class="sourceCode" id="cb97"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb97-1"><a href="texas-protocol.html#cb97-1"></a><span class="ex">python3</span> mebs_vis.py Phylo_5.tsv</span></code></pre></div>
</div>
<div id="metabolic" class="section level2" number="6.22">
<h2><span class="header-section-number">6.22</span> Metabolic</h2>
<p><a href="https://github.com/AnantharamanLab/METABOLIC">Githib</a></p>
<p>Metabolic was run from the Student folder, with the following command within the folder with the FAA files</p>
<div class="sourceCode" id="cb98"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb98-1"><a href="texas-protocol.html#cb98-1"></a><span class="fu">perl</span> /home/students2020/Tools/METABOLIC/METABOLiC-G.pl -in-gn /home/students2020/karine/folderwithFAA -o output/ -m /home/students2020/Tools/METABOLIC</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="references.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["My research.pdf", "My research.epub"],
"toc": {
"collapse": "subsection",
"edit": "https://github.com/rstudio/bookdown-demo/edit/master/%s"
}
});
});
</script>

</body>

</html>
